Watching for file changes with StatReloader
"GET / HTTP/1.1" 200 9417
"GET /static/images/chart-preview.png HTTP/1.1" 404 1886
"GET /static/images/dashboard.png HTTP/1.1" 404 1874
"GET /authentication/login/ HTTP/1.1" 200 3861
"POST /authentication/login/ HTTP/1.1" 302 0
"GET /farmer/dashboard/ HTTP/1.1" 200 6778
"GET /farmer/predictions/ HTTP/1.1" 200 11870
"POST /farmer/predictions/ HTTP/1.1" 200 12071
"POST /farmer/predictions/ HTTP/1.1" 200 12067
"GET /farmer/predictions/ HTTP/1.1" 200 11870
"GET /farmer/predictions/ HTTP/1.1" 200 11870
"GET /farmer/predictions/ HTTP/1.1" 200 11870
"GET /farmer/predictions/ HTTP/1.1" 200 11870
"GET /farmer/predictions/ HTTP/1.1" 200 11870
"POST /farmer/predictions/ HTTP/1.1" 200 12071
"GET /farmer/predictions/ HTTP/1.1" 200 11870
"GET /farmer/predictions/ HTTP/1.1" 200 11108
"GET /farmer/predictions/ HTTP/1.1" 200 11500
C:\Users\user\Desktop\soils-sense\apps\farmer\urls.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\user\Desktop\soils-sense\apps\farmer\urls.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\user\Desktop\soils-sense\apps\farmer\urls.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\user\Desktop\soils-sense\apps\farmer\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\user\Desktop\soils-sense\apps\farmer\urls.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\user\Desktop\soils-sense\apps\farmer\urls.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\user\Desktop\soils-sense\apps\farmer\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\user\Desktop\soils-sense\apps\farmer\views.py changed, reloading.
Watching for file changes with StatReloader
Watching for file changes with StatReloader
"GET /farmer/predictions/ HTTP/1.1" 200 11500
"GET /farmer/predictions/ HTTP/1.1" 200 11500
"GET /farmer/predictions/ HTTP/1.1" 200 11500
"GET /farmer/predictions/ HTTP/1.1" 200 11500
"GET / HTTP/1.1" 200 9417
"GET /static/images/chart-preview.png HTTP/1.1" 404 1886
"GET /static/images/dashboard.png HTTP/1.1" 404 1874
"GET /authentication/login/ HTTP/1.1" 200 3861
"POST /authentication/login/ HTTP/1.1" 302 0
"GET /farmer/dashboard/ HTTP/1.1" 200 6429
"GET /farmer/predictions/ HTTP/1.1" 200 11500
"POST /farmer/predictions/ HTTP/1.1" 200 11701
C:\Users\user\Desktop\soils-sense\apps\farmer\views.py changed, reloading.
Watching for file changes with StatReloader
Internal Server Error: /farmer/predictions/
Traceback (most recent call last):
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 59, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\apps\farmer\views.py", line 271, in predictions
    return render(request, 'farmer/predictions.html', context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loader.py", line 15, in get_template
    return engine.get_template(template_name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\backends\django.py", line 79, in get_template
    return Template(self.engine.get_template(template_name), self)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\engine.py", line 177, in get_template
    template, origin = self.find_template(template_name)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\engine.py", line 159, in find_template
    template = loader.get_template(name, skip=skip)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loaders\cached.py", line 57, in get_template
    template = super().get_template(template_name, skip)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loaders\base.py", line 28, in get_template
    return Template(
           ^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 154, in __init__
    self.nodelist = self.compile_nodelist()
                    ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 196, in compile_nodelist
    nodelist = parser.parse()
               ^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loader_tags.py", line 299, in do_extends
    nodelist = parser.parse()
               ^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loader_tags.py", line 234, in do_block
    nodelist = parser.parse(("endblock",))
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\defaulttags.py", line 862, in do_for
    nodelist_loop = parser.parse(
                    ^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 489, in parse
    raise self.error(token, e)
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 487, in parse
    filter_expression = self.compile_filter(token.contents)
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 609, in compile_filter
    return FilterExpression(token, self)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 705, in __init__
    filter_func = parser.find_filter(filter_name)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 615, in find_filter
    raise TemplateSyntaxError("Invalid filter: '%s'" % filter_name)
django.template.exceptions.TemplateSyntaxError: Invalid filter: 'replace'
"POST /farmer/predictions/ HTTP/1.1" 500 268945
Internal Server Error: /farmer/predictions/
Traceback (most recent call last):
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\defaulttags.py", line 1035, in find_library
    return parser.libraries[name]
           ~~~~~~~~~~~~~~~~^^^^^^
KeyError: 'farmer_extras'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 59, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\apps\farmer\views.py", line 271, in predictions
    return render(request, 'farmer/predictions.html', context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loader.py", line 15, in get_template
    return engine.get_template(template_name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\backends\django.py", line 79, in get_template
    return Template(self.engine.get_template(template_name), self)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\engine.py", line 177, in get_template
    template, origin = self.find_template(template_name)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\engine.py", line 159, in find_template
    template = loader.get_template(name, skip=skip)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loaders\cached.py", line 57, in get_template
    template = super().get_template(template_name, skip)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loaders\base.py", line 28, in get_template
    return Template(
           ^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 154, in __init__
    self.nodelist = self.compile_nodelist()
                    ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 196, in compile_nodelist
    nodelist = parser.parse()
               ^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loader_tags.py", line 299, in do_extends
    nodelist = parser.parse()
               ^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\defaulttags.py", line 1097, in load
    lib = find_library(parser, name)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\defaulttags.py", line 1037, in find_library
    raise TemplateSyntaxError(
django.template.exceptions.TemplateSyntaxError: 'farmer_extras' is not a registered tag library. Must be one of:
admin_list
admin_modify
admin_urls
cache
i18n
jazzmin
l10n
log
static
tz
"POST /farmer/predictions/ HTTP/1.1" 500 230111
"POST /farmer/predictions/ HTTP/1.1" 200 13413
"POST /farmer/predictions/ HTTP/1.1" 200 13418
"POST /farmer/predictions/ HTTP/1.1" 200 13409
"POST /farmer/predictions/ HTTP/1.1" 200 13409
"POST /farmer/predictions/ HTTP/1.1" 200 13409
"POST /farmer/predictions/ HTTP/1.1" 200 13409
"GET /authentication/logout/ HTTP/1.1" 302 0
"GET /landing_page/ HTTP/1.1" 200 9417
"GET /static/images/dashboard.png HTTP/1.1" 404 1874
"GET /static/images/chart-preview.png HTTP/1.1" 404 1886
"GET /authentication/login/ HTTP/1.1" 200 3861
"POST /authentication/login/ HTTP/1.1" 302 0
"GET /technician/dashboard/ HTTP/1.1" 200 24003
"GET /technician/reports/ HTTP/1.1" 200 17942
C:\Users\user\Desktop\soils-sense\apps\technician\views.py changed, reloading.
Watching for file changes with StatReloader
"GET /authentication/logout/ HTTP/1.1" 302 0
"GET /landing_page/ HTTP/1.1" 200 9417
"GET /static/images/dashboard.png HTTP/1.1" 404 1874
"GET /static/images/chart-preview.png HTTP/1.1" 404 1886
"GET /authentication/login/ HTTP/1.1" 200 3861
"POST /authentication/login/ HTTP/1.1" 302 0
"GET /technician/dashboard/ HTTP/1.1" 200 24003
"GET /technician/reports/ HTTP/1.1" 200 18783
C:\Users\user\Desktop\soils-sense\apps\technician\views.py changed, reloading.
Watching for file changes with StatReloader
"GET /technician/reports/ HTTP/1.1" 200 22928
"GET /technician/reports/ HTTP/1.1" 200 22770
"GET /technician/reports/ HTTP/1.1" 200 22770
"GET /technician/reports/ HTTP/1.1" 200 22770
"GET /technician/reports/ HTTP/1.1" 200 22770
"GET /technician/reports/ HTTP/1.1" 200 22770
"GET /technician/reports/ HTTP/1.1" 200 22770
"GET /technician/reports/ HTTP/1.1" 200 22770
"GET /technician/reports/ HTTP/1.1" 200 22770
"GET /technician/reports/ HTTP/1.1" 200 22770
"GET /technician/reports/ HTTP/1.1" 200 22770
"GET /technician/reports/ HTTP/1.1" 200 22770
"GET /technician/report/4/delete/ HTTP/1.1" 200 11022
"POST /technician/report/4/delete/ HTTP/1.1" 302 0
"GET /technician/reports/ HTTP/1.1" 200 21417
Internal Server Error: /technician/reports/
Traceback (most recent call last):
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\apps\technician\views.py", line 298, in reports
    soil_pred = ml.predict_soil_moisture(
                ^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: MLEngine.predict_soil_moisture() got an unexpected keyword argument 'algorithm'
"POST /technician/reports/ HTTP/1.1" 500 75444
C:\Users\user\Desktop\soils-sense\apps\technician\views.py changed, reloading.
Watching for file changes with StatReloader
Internal Server Error: /technician/reports/
Traceback (most recent call last):
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\apps\technician\views.py", line 312, in reports
    ir_pred = ml.recommend_irrigation(
              ^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: MLEngine.recommend_irrigation() got an unexpected keyword argument 'algorithm'
"POST /technician/reports/ HTTP/1.1" 500 80939
C:\Users\user\Desktop\soils-sense\apps\technician\views.py changed, reloading.
Watching for file changes with StatReloader
"POST /technician/reports/ HTTP/1.1" 302 0
"GET /technician/reports/ HTTP/1.1" 200 22763
"GET /technician/report/5/download-pdf/ HTTP/1.1" 200 1903
Internal Server Error: /technician/reports/
Traceback (most recent call last):
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\apps\technician\views.py", line 373, in reports
    ir_pred = ml.recommend_irrigation(
              ^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: MLEngine.recommend_irrigation() got an unexpected keyword argument 'algorithm'
"POST /technician/reports/ HTTP/1.1" 500 75528
C:\Users\user\Desktop\soils-sense\apps\technician\views.py changed, reloading.
Watching for file changes with StatReloader
"POST /technician/reports/ HTTP/1.1" 302 0
"GET /technician/reports/ HTTP/1.1" 200 24297
"GET /technician/report/6/download-pdf/ HTTP/1.1" 200 1871
C:\Users\user\Desktop\soils-sense\apps\technician\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\user\Desktop\soils-sense\apps\technician\views.py changed, reloading.
Watching for file changes with StatReloader
"GET /technician/report/5/delete/ HTTP/1.1" 200 11015
"POST /technician/report/5/delete/ HTTP/1.1" 302 0
"GET /technician/reports/ HTTP/1.1" 200 22766
"POST /technician/reports/ HTTP/1.1" 302 0
"GET /technician/reports/ HTTP/1.1" 200 24297
"GET /technician/report/7/download-pdf/ HTTP/1.1" 200 1881
"GET /authentication/logout/ HTTP/1.1" 302 0
"GET /landing_page/ HTTP/1.1" 200 9417
"GET /static/images/dashboard.png HTTP/1.1" 404 1874
"GET /static/images/chart-preview.png HTTP/1.1" 404 1886
"GET /authentication/signup/ HTTP/1.1" 200 5130
"GET /authentication/login/ HTTP/1.1" 200 4360
"GET /authentication/login/ HTTP/1.1" 200 3861
"POST /authentication/login/ HTTP/1.1" 302 0
"GET /technician/dashboard/ HTTP/1.1" 200 24565
"GET /technician/reports/ HTTP/1.1" 200 24297
"GET /technician/report/7/edit/ HTTP/1.1" 200 12781
"GET /technician/reports/ HTTP/1.1" 200 24297
Watching for file changes with StatReloader
"GET / HTTP/1.1" 200 9417
"GET /static/images/slide1.jpg HTTP/1.1" 304 0
"GET /static/images/dashboard.png HTTP/1.1" 404 1874
"GET /static/images/chart-preview.png HTTP/1.1" 404 1886
Not Found: /favicon.ico
"GET /favicon.ico HTTP/1.1" 404 3911
"GET /authentication/signup/ HTTP/1.1" 200 5130
"GET /static/images/logo.webp HTTP/1.1" 304 0
"GET /authentication/login/ HTTP/1.1" 200 3861
"POST /authentication/login/ HTTP/1.1" 302 0
"GET /technician/dashboard/ HTTP/1.1" 200 24565
"GET /technician/reports/ HTTP/1.1" 200 24297
"POST /technician/reports/ HTTP/1.1" 302 0
"GET /technician/reports/ HTTP/1.1" 200 25827
"GET /technician/report/8/download-pdf/ HTTP/1.1" 200 1874
"POST /technician/reports/ HTTP/1.1" 302 0
"GET /technician/reports/ HTTP/1.1" 200 27358
"GET /technician/report/9/edit/ HTTP/1.1" 200 12777
"GET /technician/reports/ HTTP/1.1" 200 27358
"GET /technician/soil-readings/ HTTP/1.1" 200 11663
"GET /technician/soil-readings/ HTTP/1.1" 200 11663
"GET /technician/farm-locations/ HTTP/1.1" 200 22788
"GET /technician/soil-readings/ HTTP/1.1" 200 11663
"GET /technician/dashboard/ HTTP/1.1" 200 25123
"GET /technician/reports/ HTTP/1.1" 200 27358
"GET /technician/reports/ HTTP/1.1" 200 27358
"GET /technician/analytics/ HTTP/1.1" 200 16648
"GET /technician/reports/ HTTP/1.1" 200 27358
"GET /technician/profile/ HTTP/1.1" 200 19406
"GET /media/users/profile_images/bruno.png HTTP/1.1" 304 0
"GET /technician/analytics/ HTTP/1.1" 200 16648
"GET /technician/farm-locations/ HTTP/1.1" 200 22788
C:\Users\user\Desktop\soils-sense\apps\technician\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\user\Desktop\soils-sense\apps\farmer\models.py changed, reloading.
Watching for file changes with StatReloader
"GET /authentication/logout/ HTTP/1.1" 302 0
"GET /landing_page/ HTTP/1.1" 200 9417
"GET /static/images/dashboard.png HTTP/1.1" 404 1874
"GET /static/images/chart-preview.png HTTP/1.1" 404 1886
"GET /authentication/login/ HTTP/1.1" 200 4083
"POST /authentication/login/ HTTP/1.1" 302 0
"GET /farmer/dashboard/ HTTP/1.1" 200 6429
"GET /farmer/soil-data/ HTTP/1.1" 200 28072
"GET /farmer/farm-management/ HTTP/1.1" 200 27163
"GET /farmer/predictions/ HTTP/1.1" 200 13212
"GET /farmer/farm-management/ HTTP/1.1" 200 27163
"POST /farmer/add-farm/ HTTP/1.1" 302 0
"GET /farmer/farm-management/ HTTP/1.1" 200 28563
"GET /farmer/soil-data/ HTTP/1.1" 200 28360
"GET /farmer/farm-management/ HTTP/1.1" 200 28243
"GET /farmer/predictions/ HTTP/1.1" 200 13212
"GET /farmer/analytics/ HTTP/1.1" 200 13152
"GET /farmer/profile/ HTTP/1.1" 200 13249
"GET /farmer/dashboard/ HTTP/1.1" 200 12995
"GET /farmer/soil-data/ HTTP/1.1" 200 28360
C:\Users\user\Desktop\soils-sense\apps\farmer\views.py changed, reloading.
Watching for file changes with StatReloader
"GET /farmer/soil-data/ HTTP/1.1" 200 29952
"GET /farmer/dashboard/ HTTP/1.1" 200 12867
"GET /farmer/farm-management/ HTTP/1.1" 200 28975
"GET /farmer/soil-data/ HTTP/1.1" 200 29952
"POST /farmer/soil-data/add/ HTTP/1.1" 200 103
"POST /farmer/soil-data/add/ HTTP/1.1" 200 103
"POST /farmer/soil-data/add/ HTTP/1.1" 200 103
"POST /farmer/soil-data/add/ HTTP/1.1" 200 103
"POST /farmer/soil-data/add/ HTTP/1.1" 200 103
"POST /farmer/soil-data/add/ HTTP/1.1" 200 103
"POST /farmer/soil-data/add/ HTTP/1.1" 200 103
"POST /farmer/soil-data/add/ HTTP/1.1" 200 103
"POST /farmer/soil-data/add/ HTTP/1.1" 200 103
"POST /farmer/soil-data/add/ HTTP/1.1" 200 58
"GET /farmer/soil-data/ HTTP/1.1" 200 31144
"GET /farmer/predictions/ HTTP/1.1" 200 16775
"GET /farmer/soil-data/ HTTP/1.1" 200 31144
"GET /farmer/predictions/ HTTP/1.1" 200 16775
"GET /farmer/get-latest-readings/3/ HTTP/1.1" 200 200
"POST /farmer/predictions/ HTTP/1.1" 302 0
"GET /farmer/predictions/ HTTP/1.1" 200 20329
"GET /farmer/get-latest-readings/3/ HTTP/1.1" 200 200
"POST /farmer/predictions/ HTTP/1.1" 302 0
"GET /farmer/predictions/ HTTP/1.1" 200 23518
"GET /farmer/get-latest-readings/3/ HTTP/1.1" 200 200
"POST /farmer/predictions/ HTTP/1.1" 302 0
"GET /farmer/predictions/ HTTP/1.1" 200 26726
"POST /farmer/predictions/delete/3/ HTTP/1.1" 302 0
"GET /farmer/predictions/ HTTP/1.1" 200 23518
"GET /farmer/predictions/download/2/ HTTP/1.1" 200 2739
"GET /farmer/profile/ HTTP/1.1" 200 12619
"GET /farmer/profile/?edit=true HTTP/1.1" 200 14488
"GET /farmer/recommendations/ HTTP/1.1" 200 12132
"GET /farmer/recommendations/?farm_id=3&algorithm=random_forest HTTP/1.1" 200 12132
"GET /farmer/recommendations/?farm_id=3&algorithm=mlp HTTP/1.1" 200 12025
"GET /farmer/recommendations/?farm_id=3&algorithm=svr HTTP/1.1" 200 12025
"GET /farmer/recommendations/?farm_id=3&algorithm=linear_regression HTTP/1.1" 200 12025
"GET /farmer/recommendations/?farm_id=3&algorithm=random_forest HTTP/1.1" 200 12132
"GET /farmer/analytics/ HTTP/1.1" 200 17253
"GET /farmer/dashboard/ HTTP/1.1" 200 12873
"GET /farmer/farm-management/ HTTP/1.1" 200 28975
"GET /farmer/dashboard/ HTTP/1.1" 200 12873
"GET /farmer/farm-management/ HTTP/1.1" 200 28975
"GET /farmer/predictions/ HTTP/1.1" 200 22993
"GET /farmer/get-latest-readings/3/ HTTP/1.1" 200 200
"GET /farmer/analytics/ HTTP/1.1" 200 17253
"GET /farmer/recommendations/ HTTP/1.1" 200 12132
"GET /farmer/profile/ HTTP/1.1" 200 12619
"GET /farmer/recommendations/ HTTP/1.1" 200 12132
"GET /farmer/recommendations/?farm_id=3&algorithm=random_forest HTTP/1.1" 200 12132
"GET /authentication/logout/ HTTP/1.1" 302 0
"GET /landing_page/ HTTP/1.1" 200 9417
"GET /static/images/chart-preview.png HTTP/1.1" 404 1886
"GET /static/images/dashboard.png HTTP/1.1" 404 1874
"GET /authentication/login/ HTTP/1.1" 200 3861
"POST /authentication/login/ HTTP/1.1" 302 0
"GET /technician/dashboard/ HTTP/1.1" 200 25560
"GET /technician/farm-locations/ HTTP/1.1" 200 26041
"GET /technician/soil-readings/ HTTP/1.1" 200 12527
"GET /technician/sensor-config/ HTTP/1.1" 200 16758
"GET /technician/analytics/ HTTP/1.1" 200 16676
"GET /technician/reports/ HTTP/1.1" 200 27402
"GET /technician/profile/ HTTP/1.1" 200 19406
"GET /technician/settings/ HTTP/1.1" 200 16943
"GET /technician/profile/ HTTP/1.1" 200 19406
"GET /technician/reports/ HTTP/1.1" 200 27402
"GET /technician/analytics/ HTTP/1.1" 200 16676
"GET /technician/sensor-config/ HTTP/1.1" 200 16758
"GET /technician/soil-readings/ HTTP/1.1" 200 12527
"GET /technician/soil-readings/ HTTP/1.1" 200 12527
"GET /technician/farm-locations/ HTTP/1.1" 200 26041
"GET /technician/dashboard/ HTTP/1.1" 200 25560
"GET /technician/analytics/ HTTP/1.1" 200 16676
"GET /technician/farm-locations/ HTTP/1.1" 200 26041
"GET /technician/reports/ HTTP/1.1" 200 27402
"GET /technician/farm-locations/ HTTP/1.1" 200 26041
"GET /technician/farm/1/ HTTP/1.1" 200 15590
"GET /technician/farm/1/edit/ HTTP/1.1" 200 14942
"GET /technician/soil-readings/?farm=1 HTTP/1.1" 200 11707
"GET /technician/farm/1/edit/ HTTP/1.1" 200 14942
"GET /technician/farm/1/ HTTP/1.1" 200 15590
"GET /technician/sensor-config/ HTTP/1.1" 200 16758
"GET /technician/analytics/ HTTP/1.1" 200 16676
"GET /technician/soil-readings/ HTTP/1.1" 200 12527
"GET /technician/soil-readings/ HTTP/1.1" 200 12527
"GET /technician/soil-readings/ HTTP/1.1" 200 12527
"GET /technician/sensor-config/ HTTP/1.1" 200 16758
"GET /technician/sensor-config/ HTTP/1.1" 200 16758
"GET /technician/sensor-config/ HTTP/1.1" 200 16758
"GET /technician/analytics/ HTTP/1.1" 200 16676
"GET /technician/reports/ HTTP/1.1" 200 27402
"GET /technician/profile/ HTTP/1.1" 200 19406
"GET /authentication/reset/ HTTP/1.1" 200 3254
"GET /authentication/reset/ HTTP/1.1" 200 3254
"POST /authentication/reset/ HTTP/1.1" 302 0
"GET /authentication/reset/done/ HTTP/1.1" 200 2346
"GET /technician/settings/ HTTP/1.1" 200 16943
"GET /technician/profile/ HTTP/1.1" 200 19406
"GET /technician/reports/ HTTP/1.1" 200 27402
"GET /technician/sensor-config/ HTTP/1.1" 200 16758
"GET /technician/soil-readings/ HTTP/1.1" 200 12527
"GET /technician/analytics/ HTTP/1.1" 200 16676
"GET /technician/reports/ HTTP/1.1" 200 27402
"GET /technician/sensor-config/ HTTP/1.1" 200 16758
C:\Users\user\Desktop\soils-sense\apps\technician\urls.py changed, reloading.
Watching for file changes with StatReloader
Not Found: /technician/sensor-config/
"GET /technician/sensor-config/ HTTP/1.1" 404 9154
"GET /technician/reports/ HTTP/1.1" 200 27384
"GET /technician/reports/ HTTP/1.1" 200 27384
Internal Server Error: /technician/models/
Traceback (most recent call last):
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\apps\technician\views.py", line 678, in models_view
    ml_predictor = MLEngine.SoilMoisturePredictor()
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: type object 'MLEngine' has no attribute 'SoilMoisturePredictor'
- Broken pipe from ('127.0.0.1', 60163)
Internal Server Error: /technician/models/
Traceback (most recent call last):
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\apps\technician\views.py", line 678, in models_view
    ml_predictor = MLEngine.SoilMoisturePredictor()
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: type object 'MLEngine' has no attribute 'SoilMoisturePredictor'
"GET /technician/models/ HTTP/1.1" 500 68413
C:\Users\user\Desktop\soils-sense\apps\technician\views.py changed, reloading.
Watching for file changes with StatReloader
Internal Server Error: /technician/models/
Traceback (most recent call last):
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\apps\technician\views.py", line 679, in models_view
    models_list = ml_predictor.get_available_models()
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'SoilMoisturePredictor' object has no attribute 'get_available_models'
"GET /technician/models/ HTTP/1.1" 500 68618
Internal Server Error: /technician/models/
Traceback (most recent call last):
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\apps\technician\views.py", line 679, in models_view
    models_list = ml_predictor.get_available_models()
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'SoilMoisturePredictor' object has no attribute 'get_available_models'
"GET /technician/models/ HTTP/1.1" 500 68618
Internal Server Error: /technician/models/
Traceback (most recent call last):
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\apps\technician\views.py", line 679, in models_view
    models_list = ml_predictor.get_available_models()
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'SoilMoisturePredictor' object has no attribute 'get_available_models'
"GET /technician/models/ HTTP/1.1" 500 68481
C:\Users\user\Desktop\soils-sense\apps\technician\views.py changed, reloading.
Watching for file changes with StatReloader
"GET /technician/models/ HTTP/1.1" 200 11198
C:\Users\user\Desktop\soils-sense\apps\technician\views.py changed, reloading.
Watching for file changes with StatReloader
"GET /technician/models/ HTTP/1.1" 200 14789
"GET /technician/analytics/ HTTP/1.1" 200 16658
"GET /technician/reports/ HTTP/1.1" 200 27384
Not Found: /farmer
"GET /farmer HTTP/1.1" 404 3896
Watching for file changes with StatReloader
Watching for file changes with StatReloader
"GET / HTTP/1.1" 200 9417
"GET /static/images/dashboard.png HTTP/1.1" 404 1874
"GET /static/images/chart-preview.png HTTP/1.1" 404 1886
"GET /authentication/login/ HTTP/1.1" 200 3861
"POST /authentication/login/ HTTP/1.1" 302 0
"GET /farmer/dashboard/ HTTP/1.1" 200 12873
"GET /farmer/predictions/ HTTP/1.1" 200 22993
C:\Users\user\Desktop\soils-sense\apps\technician\views.py changed, reloading.
Watching for file changes with StatReloader
"GET /technician/reports/ HTTP/1.1" 200 22954
"GET /technician/farm-locations/ HTTP/1.1" 200 26023
"GET /technician/farm-locations/ HTTP/1.1" 200 26023
"GET /technician/farm-locations/ HTTP/1.1" 200 26023
"GET /technician/farm-locations/ HTTP/1.1" 200 26023
"GET /technician/reports/ HTTP/1.1" 200 22954
"GET /technician/reports/ HTTP/1.1" 200 23087
"GET /technician/reports/ HTTP/1.1" 200 23087
"GET /technician/farm-locations/ HTTP/1.1" 200 26156
"GET /technician/farm/1/ HTTP/1.1" 200 15705
"GET /farmer/farm-management/ HTTP/1.1" 200 28975
C:\Users\user\Desktop\soils-sense\apps\technician\views.py changed, reloading.
Watching for file changes with StatReloader
"GET /technician/farm-locations/ HTTP/1.1" 200 26023
"GET /technician/farm-locations/ HTTP/1.1" 200 26023
"GET /technician/farm-locations/ HTTP/1.1" 200 26023
"GET /technician/farm-locations/ HTTP/1.1" 200 26023
Internal Server Error: /technician/dashboard/
Traceback (most recent call last):
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\apps\technician\views.py", line 133, in dashboard
    return render(request, 'technician/dashboard.html', context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\backends\django.py", line 107, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loader_tags.py", line 65, in render
    result = block.nodelist.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\defaulttags.py", line 480, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\urls\base.py", line 98, in reverse
    resolved_url = resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\urls\resolvers.py", line 831, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'sensor_config' not found. 'sensor_config' is not a valid view function or pattern name.
"GET /technician/dashboard/ HTTP/1.1" 500 202125
Internal Server Error: /technician/dashboard/
Traceback (most recent call last):
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\apps\technician\views.py", line 133, in dashboard
    return render(request, 'technician/dashboard.html', context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\backends\django.py", line 107, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loader_tags.py", line 65, in render
    result = block.nodelist.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\defaulttags.py", line 480, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\urls\base.py", line 98, in reverse
    resolved_url = resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\urls\resolvers.py", line 831, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'sensor_config' not found. 'sensor_config' is not a valid view function or pattern name.
"GET /technician/dashboard/ HTTP/1.1" 500 201948
"GET / HTTP/1.1" 200 9417
"GET /static/images/dashboard.png HTTP/1.1" 404 1874
"GET /static/images/chart-preview.png HTTP/1.1" 404 1886
"GET /authentication/login/ HTTP/1.1" 200 3861
"POST /authentication/login/ HTTP/1.1" 302 0
Internal Server Error: /technician/dashboard/
Traceback (most recent call last):
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\apps\technician\views.py", line 133, in dashboard
    return render(request, 'technician/dashboard.html', context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\backends\django.py", line 107, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loader_tags.py", line 65, in render
    result = block.nodelist.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\defaulttags.py", line 480, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\urls\base.py", line 98, in reverse
    resolved_url = resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\urls\resolvers.py", line 831, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'sensor_config' not found. 'sensor_config' is not a valid view function or pattern name.
"GET /technician/dashboard/ HTTP/1.1" 500 202259
"GET /technician/dashboard/ HTTP/1.1" 200 25526
"GET /technician/farm-locations/ HTTP/1.1" 200 26023
"GET /technician/farm-locations/ HTTP/1.1" 200 26023
"GET /technician/farm/1/ HTTP/1.1" 200 15572
"GET /technician/farm-locations/ HTTP/1.1" 200 19448
"GET /technician/farm/1/ HTTP/1.1" 200 15572
"GET /technician/farm/1/edit/ HTTP/1.1" 200 14924
"GET /technician/farm-locations/ HTTP/1.1" 200 19448
"GET /technician/farm-locations/ HTTP/1.1" 200 18747
"GET /technician/farm-locations/ HTTP/1.1" 200 18593
"GET /technician/farm-locations/ HTTP/1.1" 200 16251
"GET /technician/soil-readings/ HTTP/1.1" 200 12509
"GET /technician/soil-readings/ HTTP/1.1" 200 12269
"GET /technician/soil-readings/?farm=3&date_from=&date_to=&status= HTTP/1.1" 200 12287
"GET /technician/soil-readings/?farm=3&date_from=&date_to=&status= HTTP/1.1" 200 12287
"GET /farmer/soil-data/ HTTP/1.1" 200 30294
"GET /technician/reports/ HTTP/1.1" 200 22954
"GET /farmer/farm-management/ HTTP/1.1" 200 30817
"GET /farmer/soil-data/ HTTP/1.1" 200 30294
"POST /farmer/soil-data/add/ HTTP/1.1" 200 58
"GET /farmer/soil-data/ HTTP/1.1" 200 31481
C:\Users\user\Desktop\soils-sense\apps\technician\views.py changed, reloading.
Watching for file changes with StatReloader
"GET /farmer/predictions/ HTTP/1.1" 200 16666
"GET /farmer/farm-management/ HTTP/1.1" 200 27895
"POST /farmer/add-farm/ HTTP/1.1" 302 0
"GET /farmer/farm-management/ HTTP/1.1" 200 29306
"GET /farmer/soil-data/ HTTP/1.1" 200 29958
"POST /farmer/soil-data/add/ HTTP/1.1" 200 58
"GET /farmer/soil-data/ HTTP/1.1" 200 31145
"GET /farmer/predictions/ HTTP/1.1" 200 16776
"GET /farmer/get-latest-readings/4/ HTTP/1.1" 200 203
"POST /farmer/predictions/ HTTP/1.1" 302 0
"GET /farmer/predictions/ HTTP/1.1" 200 20330
"GET /technician/reports/ HTTP/1.1" 200 20369
"GET /technician/reports/ HTTP/1.1" 200 21021
"GET /technician/reports/ HTTP/1.1" 200 15870
"GET /technician/reports/ HTTP/1.1" 200 16332
"GET /technician/reports/ HTTP/1.1" 200 15865
"GET /technician/reports/ HTTP/1.1" 200 16875
- Broken pipe from ('127.0.0.1', 62374)
"GET /technician/models/ HTTP/1.1" 200 14789
- Broken pipe from ('127.0.0.1', 63401)
"GET /technician/models/ HTTP/1.1" 200 14789
"GET /technician/farm-locations/ HTTP/1.1" 200 14112
"GET /technician/farm/4/ HTTP/1.1" 200 15867
"GET /technician/dashboard/ HTTP/1.1" 200 23196
- Broken pipe from ('127.0.0.1', 63403)
"GET /technician/models/ HTTP/1.1" 200 14789
"GET /technician/analytics/ HTTP/1.1" 200 16659
"GET /technician/reports/ HTTP/1.1" 200 16894
"GET /technician/reports/ HTTP/1.1" 200 16894
"GET /technician/reports/ HTTP/1.1" 200 16894
"GET /technician/reports/ HTTP/1.1" 200 16894
"GET /technician/dashboard/ HTTP/1.1" 200 23196
"GET /technician/reports/ HTTP/1.1" 200 16894
"GET /technician/reports/ HTTP/1.1" 200 16875
"GET /technician/reports/ HTTP/1.1" 200 16875
"GET /technician/reports/ HTTP/1.1" 200 16875
"GET /technician/dashboard/ HTTP/1.1" 200 23196
"GET /technician/reports/ HTTP/1.1" 200 16875
"GET /technician/reports/ HTTP/1.1" 200 16875
"GET /technician/reports/ HTTP/1.1" 200 16894
"GET /technician/reports/ HTTP/1.1" 200 16894
"GET /technician/reports/ HTTP/1.1" 200 16894
"GET /technician/reports/ HTTP/1.1" 200 16894
"GET /technician/reports/ HTTP/1.1" 200 16894
"GET /technician/reports/ HTTP/1.1" 200 16894
"GET /technician/reports/ HTTP/1.1" 200 16894
"GET /technician/reports/ HTTP/1.1" 200 16894
"GET /technician/reports/ HTTP/1.1" 200 16894
"GET /technician/reports/ HTTP/1.1" 200 16894
"GET /technician/reports/ HTTP/1.1" 200 16894
"GET /authentication/logout/ HTTP/1.1" 302 0
"GET /landing_page/ HTTP/1.1" 200 9417
"GET /static/images/dashboard.png HTTP/1.1" 404 1874
"GET /static/images/chart-preview.png HTTP/1.1" 404 1886
"GET /authentication/login/ HTTP/1.1" 200 3861
"POST /authentication/login/ HTTP/1.1" 302 0
"GET /technician/dashboard/ HTTP/1.1" 200 23196
"GET /technician/reports/ HTTP/1.1" 200 16894
"GET /technician/reports/ HTTP/1.1" 200 22045
Internal Server Error: /technician/reports/
Traceback (most recent call last):
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 510, in parse
    compile_func = self.tags[command]
                   ~~~~~~~~~^^^^^^^^^
KeyError: 'endblock'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 59, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\apps\technician\views.py", line 328, in reports
    return render(request, 'technician/reports.html', context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loader.py", line 15, in get_template
    return engine.get_template(template_name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\backends\django.py", line 79, in get_template
    return Template(self.engine.get_template(template_name), self)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\engine.py", line 177, in get_template
    template, origin = self.find_template(template_name)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\engine.py", line 159, in find_template
    template = loader.get_template(name, skip=skip)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loaders\cached.py", line 57, in get_template
    template = super().get_template(template_name, skip)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loaders\base.py", line 28, in get_template
    return Template(
           ^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 154, in __init__
    self.nodelist = self.compile_nodelist()
                    ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 196, in compile_nodelist
    nodelist = parser.parse()
               ^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loader_tags.py", line 299, in do_extends
    nodelist = parser.parse()
               ^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 512, in parse
    self.invalid_block_tag(token, command, parse_until)
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 577, in invalid_block_tag
    raise self.error(
django.template.exceptions.TemplateSyntaxError: Invalid block tag on line 278: 'endblock'. Did you forget to register or load this tag?
"GET /technician/reports/ HTTP/1.1" 500 188811
"GET /technician/reports/ HTTP/1.1" 200 21020
"GET /technician/reports/ HTTP/1.1" 200 23113
"GET /technician/reports/ HTTP/1.1" 200 21020
- Broken pipe from ('127.0.0.1', 63484)
"GET /technician/models/ HTTP/1.1" 200 14789
"GET /technician/reports/ HTTP/1.1" 200 22824
"GET /technician/reports/ HTTP/1.1" 200 22824
"GET /technician/reports/ HTTP/1.1" 200 15883
"GET /technician/reports/ HTTP/1.1" 200 16442
"GET /technician/dashboard/ HTTP/1.1" 200 23196
"GET /technician/farm-locations/ HTTP/1.1" 200 14112
"GET /technician/soil-readings/ HTTP/1.1" 200 12183
"GET /technician/models/ HTTP/1.1" 200 14789
"GET /technician/analytics/ HTTP/1.1" 200 16659
"GET /technician/reports/ HTTP/1.1" 200 15883
"GET /technician/reports/ HTTP/1.1" 200 15953
"GET /farmer/get-latest-readings/4/ HTTP/1.1" 200 203
"GET /farmer/farm-management/ HTTP/1.1" 200 28985
"GET /farmer/predictions/ HTTP/1.1" 200 19805
"GET /farmer/soil-data/ HTTP/1.1" 200 31145
"GET /farmer/analytics/ HTTP/1.1" 200 17254
"GET /farmer/recommendations/ HTTP/1.1" 200 12133
"GET /farmer/farm-management/ HTTP/1.1" 200 28985
"GET /technician/reports/ HTTP/1.1" 200 15976
"GET /technician/reports/ HTTP/1.1" 200 15976
"GET /technician/farm-locations/ HTTP/1.1" 200 14112
"GET /technician/reports/ HTTP/1.1" 200 15976
"GET /technician/reports/ HTTP/1.1" 200 15953
"GET /technician/reports/ HTTP/1.1" 200 15873
"GET /technician/reports/ HTTP/1.1" 200 16342
"GET /authentication/logout/ HTTP/1.1" 302 0
"GET /landing_page/ HTTP/1.1" 200 9417
"GET /static/images/dashboard.png HTTP/1.1" 404 1874
"GET /static/images/chart-preview.png HTTP/1.1" 404 1886
"GET /authentication/login/ HTTP/1.1" 200 3861
"POST /authentication/login/ HTTP/1.1" 302 0
"GET /technician/dashboard/ HTTP/1.1" 200 23679
"GET /technician/dashboard/ HTTP/1.1" 200 23116
"GET /technician/farm-locations/ HTTP/1.1" 200 14032
"GET /technician/soil-readings/ HTTP/1.1" 200 12103
"GET /technician/soil-readings/ HTTP/1.1" 200 12103
"GET /technician/soil-readings/?farm=4&date_from=&date_to=&status= HTTP/1.1" 200 12121
"GET /authentication/logout/ HTTP/1.1" 302 0
"GET /landing_page/ HTTP/1.1" 200 9417
"GET /static/images/dashboard.png HTTP/1.1" 404 1874
"GET /static/images/chart-preview.png HTTP/1.1" 404 1886
"GET /authentication/login/ HTTP/1.1" 200 3861
"POST /authentication/login/ HTTP/1.1" 302 0
"GET /farmer/dashboard/ HTTP/1.1" 200 7161
"GET /technician/reports/ HTTP/1.1" 200 15873
"GET /technician/soil-readings/ HTTP/1.1" 302 0
Not Found: /accounts/login/
"GET /accounts/login/?next=/technician/soil-readings/ HTTP/1.1" 404 3955
"GET /authentication/logout/ HTTP/1.1" 302 0
"GET /landing_page/ HTTP/1.1" 200 9417
"GET /static/images/dashboard.png HTTP/1.1" 404 1874
"GET /static/images/chart-preview.png HTTP/1.1" 404 1886
"GET /authentication/login/ HTTP/1.1" 200 3861
"POST /authentication/login/ HTTP/1.1" 302 0
"GET /technician/dashboard/ HTTP/1.1" 200 23116
"GET /technician/soil-readings/ HTTP/1.1" 200 12103
"GET /farmer/predictions/ HTTP/1.1" 200 19805
"GET /farmer/soil-data/ HTTP/1.1" 200 31145
"GET /farmer/farm-management/ HTTP/1.1" 200 28985
"GET /farmer/soil-data/ HTTP/1.1" 200 31145
"GET /farmer/predictions/ HTTP/1.1" 200 19805
"GET /farmer/analytics/ HTTP/1.1" 200 17254
"GET /farmer/recommendations/ HTTP/1.1" 200 12133
"GET /farmer/analytics/ HTTP/1.1" 200 17254
"GET /farmer/predictions/ HTTP/1.1" 200 19805
"GET /farmer/recommendations/ HTTP/1.1" 200 12133
"GET /technician/soil-readings/ HTTP/1.1" 200 12026
"GET /technician/reports/ HTTP/1.1" 200 15873
"GET /technician/reports/ HTTP/1.1" 200 15873
"GET /technician/reports/ HTTP/1.1" 200 15896
"GET /technician/reports/ HTTP/1.1" 200 15896
"GET /technician/reports/ HTTP/1.1" 200 15896
"GET /technician/reports/ HTTP/1.1" 200 15896
"GET /technician/reports/ HTTP/1.1" 200 15896
"GET /technician/reports/ HTTP/1.1" 200 15896
"GET /technician/reports/ HTTP/1.1" 200 15896
"GET /technician/reports/ HTTP/1.1" 200 15896
"GET /technician/reports/ HTTP/1.1" 200 15896
"GET /technician/reports/ HTTP/1.1" 200 15896
"GET /technician/reports/ HTTP/1.1" 200 15896
"GET /technician/reports/ HTTP/1.1" 200 15896
"GET /technician/reports/ HTTP/1.1" 200 15919
"GET /technician/reports/ HTTP/1.1" 200 15919
"GET /technician/reports/ HTTP/1.1" 200 15919
"GET /technician/reports/ HTTP/1.1" 200 15919
"GET /technician/reports/ HTTP/1.1" 200 15919
"GET /technician/reports/ HTTP/1.1" 200 15919
"GET /technician/reports/ HTTP/1.1" 200 15919
"GET /technician/reports/ HTTP/1.1" 200 15919
"GET /technician/dashboard/ HTTP/1.1" 200 23116
"GET /technician/reports/ HTTP/1.1" 200 15919
"GET /technician/reports/ HTTP/1.1" 200 15919
"GET /technician/reports/ HTTP/1.1" 200 15915
"GET /technician/reports/ HTTP/1.1" 200 15919
"GET /technician/profile/ HTTP/1.1" 200 19308
"GET /technician/reports/ HTTP/1.1" 200 15919
"GET /technician/analytics/ HTTP/1.1" 200 16579
"GET /technician/analytics/ HTTP/1.1" 200 16338
"GET /technician/reports/ HTTP/1.1" 200 15919
"GET /technician/analytics/ HTTP/1.1" 200 16338
"GET /technician/farm-locations/ HTTP/1.1" 200 14032
"GET /technician/dashboard/ HTTP/1.1" 200 23116
"GET /technician/reports/ HTTP/1.1" 200 15919
"GET /technician/soil-readings/ HTTP/1.1" 200 12026
"GET /technician/reports/ HTTP/1.1" 200 15919
"GET /technician/reports/ HTTP/1.1" 200 17155
"GET /technician/reports/ HTTP/1.1" 200 17155
"GET /technician/reports/ HTTP/1.1" 200 17155
C:\Users\user\Desktop\soils-sense\apps\technician\views.py changed, reloading.
Watching for file changes with StatReloader
"GET /technician/reports/ HTTP/1.1" 200 17155
"GET /technician/reports/ HTTP/1.1" 200 17155
"GET /technician/reports/ HTTP/1.1" 200 17155
"GET /technician/reports/ HTTP/1.1" 200 17155
"GET /technician/reports/ HTTP/1.1" 200 17155
"GET /technician/reports/ HTTP/1.1" 200 17155
"GET /technician/reports/ HTTP/1.1" 200 17155
"GET /technician/dashboard/ HTTP/1.1" 200 23116
"GET /technician/dashboard/ HTTP/1.1" 200 23116
"GET /technician/reports/ HTTP/1.1" 200 17259
"GET /technician/farm-locations/ HTTP/1.1" 200 14032
"GET /technician/dashboard/ HTTP/1.1" 200 23116
"GET /technician/reports/ HTTP/1.1" 200 17259
C:\Users\user\Desktop\soils-sense\apps\farmer\apps.py changed, reloading.
Watching for file changes with StatReloader
"GET /farmer/recommendations/ HTTP/1.1" 200 12133
"GET /farmer/recommendations/ HTTP/1.1" 200 12133
"GET /farmer/farm-management/ HTTP/1.1" 200 28985
"POST /farmer/add-farm/ HTTP/1.1" 302 0
"GET /farmer/farm-management/ HTTP/1.1" 200 31126
"GET /technician/reports/ HTTP/1.1" 200 17929
C:\Users\user\Desktop\soils-sense\apps\farmer\apps.py changed, reloading.
Watching for file changes with StatReloader
"GET /technician/reports/ HTTP/1.1" 200 17929
"GET /technician/reports/ HTTP/1.1" 200 17929
C:\Users\user\Desktop\soils-sense\apps\technician\views.py changed, reloading.
Watching for file changes with StatReloader
"GET /technician/reports/ HTTP/1.1" 200 17259
"GET /technician/reports/ HTTP/1.1" 200 17259
"GET /technician/reports/ HTTP/1.1" 200 17259
"GET /technician/reports/ HTTP/1.1" 200 17259
"GET /technician/dashboard/ HTTP/1.1" 200 23533
"GET /technician/reports/ HTTP/1.1" 200 14918
"GET /technician/reports/ HTTP/1.1" 200 14918
"GET /technician/reports/ HTTP/1.1" 200 14918
"GET /technician/reports/ HTTP/1.1" 200 14918
"GET /technician/reports/ HTTP/1.1" 200 17259
"GET /technician/reports/ HTTP/1.1" 200 15803
"GET /technician/dashboard/ HTTP/1.1" 200 23533
"GET /technician/dashboard/ HTTP/1.1" 200 22447
"GET /technician/dashboard/ HTTP/1.1" 200 22501
"GET /technician/dashboard/ HTTP/1.1" 200 22545
"GET /technician/farm-locations/ HTTP/1.1" 200 15139
"GET /technician/soil-readings/ HTTP/1.1" 200 12116
"GET /technician/models/ HTTP/1.1" 200 14753
"GET /technician/analytics/ HTTP/1.1" 200 16382
"GET /technician/reports/ HTTP/1.1" 200 15847
"GET /technician/profile/ HTTP/1.1" 200 19352
"GET /technician/settings/ HTTP/1.1" 200 16889
"GET /technician/farm-locations/ HTTP/1.1" 200 15139
"GET /technician/soil-readings/ HTTP/1.1" 200 12116
"GET /technician/models/ HTTP/1.1" 200 14753
"GET /technician/analytics/ HTTP/1.1" 200 16382
"GET /technician/reports/ HTTP/1.1" 200 15847
"GET /technician/profile/ HTTP/1.1" 200 19352
"GET /technician/reports/ HTTP/1.1" 200 15847
"GET /technician/analytics/ HTTP/1.1" 200 16382
"GET /technician/models/ HTTP/1.1" 200 14753
"GET /technician/soil-readings/ HTTP/1.1" 200 12116
"GET /technician/farm-locations/ HTTP/1.1" 200 15139
"GET /technician/dashboard/ HTTP/1.1" 200 22545
"GET /technician/dashboard/ HTTP/1.1" 200 22310
"GET /technician/soil-readings/ HTTP/1.1" 200 12116
"GET /technician/farm-locations/ HTTP/1.1" 200 15139
Watching for file changes with StatReloader
"GET /farmer/soil-data/ HTTP/1.1" 200 31490
"POST /farmer/soil-data/add/ HTTP/1.1" 200 58
"GET /farmer/soil-data/ HTTP/1.1" 200 32844
"GET /farmer/predictions/ HTTP/1.1" 200 19915
"GET /farmer/get-latest-readings/5/ HTTP/1.1" 200 198
"POST /farmer/predictions/ HTTP/1.1" 302 0
"GET /farmer/predictions/ HTTP/1.1" 200 23631
"GET /farmer/analytics/ HTTP/1.1" 200 17371
"GET /technician/reports/ HTTP/1.1" 200 18962
"GET /technician/analytics/ HTTP/1.1" 200 16399
"GET /technician/reports/ HTTP/1.1" 200 18962
"GET /technician/settings/ HTTP/1.1" 200 16889
"GET /technician/profile/ HTTP/1.1" 200 19352
"GET /technician/analytics/ HTTP/1.1" 200 16399
"GET /technician/models/ HTTP/1.1" 200 14753
"GET /technician/soil-readings/ HTTP/1.1" 200 13051
"GET /technician/farm-locations/ HTTP/1.1" 200 15139
"GET /technician/dashboard/ HTTP/1.1" 200 22317
"GET /technician/reports/ HTTP/1.1" 200 18962
Not Found: /technician/report/5/download-pdf/
"GET /technician/report/5/download-pdf/ HTTP/1.1" 404 8073
Not Found: /technician/report/5/delete/
"POST /technician/report/5/delete/ HTTP/1.1" 404 7735
C:\Users\user\Desktop\soils-sense\apps\technician\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\user\Desktop\soils-sense\apps\technician\urls.py changed, reloading.
Watching for file changes with StatReloader
Not Found: /technician/report/5/download-pdf/
"GET /technician/report/5/download-pdf/ HTTP/1.1" 404 8073
C:\Users\user\Desktop\soils-sense\apps\technician\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\user\Desktop\soils-sense\apps\technician\urls.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\user\Desktop\soils-sense\apps\technician\views.py changed, reloading.
Watching for file changes with StatReloader
Not Found: /technician/report/5/download-pdf/
"GET /technician/report/5/download-pdf/ HTTP/1.1" 404 8073
Not Found: /technician/report/5/download-pdf/
"GET /technician/report/5/download-pdf/ HTTP/1.1" 404 8073
Not Found: /technician/report/4/delete/
"POST /technician/report/4/delete/ HTTP/1.1" 404 7735
"GET /authentication/logout/ HTTP/1.1" 302 0
"GET /landing_page/ HTTP/1.1" 200 9417
"GET /static/images/dashboard.png HTTP/1.1" 404 1874
"GET /static/images/chart-preview.png HTTP/1.1" 404 1886
"GET /authentication/login/ HTTP/1.1" 200 3861
"POST /authentication/login/ HTTP/1.1" 302 0
"GET /technician/dashboard/ HTTP/1.1" 200 22317
"GET /technician/reports/ HTTP/1.1" 200 18978
"POST /technician/prediction/4/delete/ HTTP/1.1" 302 0
"GET /technician/reports/ HTTP/1.1" 200 15853
"GET /technician/prediction/5/download-pdf/ HTTP/1.1" 302 0
"GET /technician/reports/ HTTP/1.1" 200 15853
"GET /technician/prediction/5/download-pdf/ HTTP/1.1" 302 0
"GET /technician/reports/ HTTP/1.1" 200 15853
"GET /technician/prediction/5/download-pdf/ HTTP/1.1" 302 0
"GET /technician/reports/ HTTP/1.1" 200 15853
"GET /technician/prediction/5/download-pdf/ HTTP/1.1" 302 0
"GET /technician/reports/ HTTP/1.1" 200 15853
"GET /technician/prediction/5/download-pdf/ HTTP/1.1" 302 0
"GET /technician/reports/ HTTP/1.1" 200 15853
"GET /technician/prediction/5/download-pdf/ HTTP/1.1" 302 0
"GET /technician/reports/ HTTP/1.1" 200 15853
"GET /technician/reports/ HTTP/1.1" 200 15853
"GET /technician/farm-locations/ HTTP/1.1" 200 15139
"GET /technician/dashboard/ HTTP/1.1" 200 22317
"GET /technician/profile/ HTTP/1.1" 200 19352
"GET /technician/analytics/ HTTP/1.1" 200 16399
"GET /technician/reports/ HTTP/1.1" 200 15853
"GET /technician/prediction/5/download-pdf/ HTTP/1.1" 302 0
"GET /technician/reports/ HTTP/1.1" 200 15853
"GET /technician/prediction/5/download-pdf/ HTTP/1.1" 302 0
"GET /technician/reports/ HTTP/1.1" 200 15853
C:\Users\user\Desktop\soils-sense\apps\technician\views.py changed, reloading.
Watching for file changes with StatReloader
"GET /technician/reports/ HTTP/1.1" 200 15853
"GET /technician/prediction/5/download-pdf/ HTTP/1.1" 302 0
"GET /technician/reports/ HTTP/1.1" 200 15853
"GET /technician/prediction/5/download-pdf/ HTTP/1.1" 302 0
"GET /technician/reports/ HTTP/1.1" 200 15853
"GET /technician/prediction/5/download-pdf/ HTTP/1.1" 302 0
"GET /technician/reports/ HTTP/1.1" 200 15853
"GET /technician/prediction/5/download-pdf/ HTTP/1.1" 302 0
"GET /technician/reports/ HTTP/1.1" 200 15853
"GET /technician/prediction/5/download-pdf/ HTTP/1.1" 302 0
"GET /technician/reports/ HTTP/1.1" 200 15853
"GET /technician/prediction/5/download-pdf/ HTTP/1.1" 302 0
"GET /technician/reports/ HTTP/1.1" 200 15853
"GET /technician/prediction/5/download-pdf/ HTTP/1.1" 302 0
"GET /technician/reports/ HTTP/1.1" 200 15853
"GET /technician/prediction/5/download-pdf/ HTTP/1.1" 302 0
"GET /technician/reports/ HTTP/1.1" 200 15853
"GET /technician/prediction/5/download-pdf/ HTTP/1.1" 302 0
"GET /technician/reports/ HTTP/1.1" 200 15853
"GET /authentication/logout/ HTTP/1.1" 302 0
"GET /landing_page/ HTTP/1.1" 200 9417
"GET /static/images/dashboard.png HTTP/1.1" 404 1874
"GET /static/images/chart-preview.png HTTP/1.1" 404 1886
"GET /authentication/signup/ HTTP/1.1" 200 5130
"GET /authentication/login/ HTTP/1.1" 200 5989
C:\Users\user\Desktop\soils-sense\apps\technician\views.py changed, reloading.
Watching for file changes with StatReloader
"GET /authentication/login/ HTTP/1.1" 200 3861
"POST /authentication/login/ HTTP/1.1" 302 0
"GET /technician/dashboard/ HTTP/1.1" 200 22317
"GET /technician/reports/ HTTP/1.1" 200 15853
"GET /technician/prediction/5/download-pdf/ HTTP/1.1" 200 2702
C:\Users\user\Desktop\soils-sense\apps\technician\views.py changed, reloading.
Watching for file changes with StatReloader
"GET /authentication/logout/ HTTP/1.1" 302 0
"GET /landing_page/ HTTP/1.1" 200 9417
"GET /static/images/dashboard.png HTTP/1.1" 404 1874
"GET /static/images/chart-preview.png HTTP/1.1" 404 1886
"GET /authentication/signup/ HTTP/1.1" 200 5130
"GET /authentication/login/ HTTP/1.1" 200 3861
"POST /authentication/login/ HTTP/1.1" 302 0
"GET /technician/dashboard/ HTTP/1.1" 200 22317
"GET /technician/reports/ HTTP/1.1" 200 15853
"GET /technician/prediction/5/download-pdf/ HTTP/1.1" 200 2732
C:\Users\user\Desktop\soils-sense\apps\technician\views.py changed, reloading.
Watching for file changes with StatReloader
"GET /authentication/logout/ HTTP/1.1" 302 0
"GET /landing_page/ HTTP/1.1" 200 9417
"GET /static/images/dashboard.png HTTP/1.1" 404 1874
"GET /static/images/chart-preview.png HTTP/1.1" 404 1886
"GET /authentication/login/ HTTP/1.1" 200 3861
"POST /authentication/login/ HTTP/1.1" 302 0
"GET /technician/dashboard/ HTTP/1.1" 200 22317
"GET /technician/reports/ HTTP/1.1" 200 15853
"GET /technician/prediction/5/download-pdf/ HTTP/1.1" 200 2733
"GET /technician/dashboard/ HTTP/1.1" 200 22317
"GET /technician/farm-locations/ HTTP/1.1" 200 15139
"GET /technician/reports/ HTTP/1.1" 200 15932
"GET /technician/dashboard/ HTTP/1.1" 200 22317
"GET /technician/dashboard/ HTTP/1.1" 200 22317
"GET /technician/models/ HTTP/1.1" 200 14753
"GET /technician/reports/ HTTP/1.1" 200 15582
"GET /technician/profile/ HTTP/1.1" 200 19352
"GET /technician/settings/ HTTP/1.1" 200 16889
"GET /technician/reports/ HTTP/1.1" 200 15582
"GET /technician/analytics/ HTTP/1.1" 200 16399
"GET /technician/models/ HTTP/1.1" 200 14753
"GET /technician/soil-readings/ HTTP/1.1" 200 13051
"GET /technician/farm-locations/ HTTP/1.1" 200 15139
"GET /technician/analytics/ HTTP/1.1" 200 16399
"GET /technician/farm-locations/ HTTP/1.1" 200 15139
"GET /technician/dashboard/ HTTP/1.1" 200 22317
"GET /technician/farm-locations/ HTTP/1.1" 200 15139
"GET /technician/soil-readings/ HTTP/1.1" 200 13051
"GET /technician/soil-readings/ HTTP/1.1" 200 13051
"GET /technician/models/ HTTP/1.1" 200 14753
"GET /technician/analytics/ HTTP/1.1" 200 16399
"GET /technician/reports/ HTTP/1.1" 200 15582
"GET /technician/dashboard/ HTTP/1.1" 200 22317
"GET /technician/soil-readings/ HTTP/1.1" 200 13051
"GET /technician/settings/ HTTP/1.1" 200 16889
"GET /media/users/profile_images/bruno.png HTTP/1.1" 304 0
"GET /authentication/logout/ HTTP/1.1" 302 0
"GET /landing_page/ HTTP/1.1" 200 9417
"GET /static/images/dashboard.png HTTP/1.1" 404 1874
"GET /static/images/chart-preview.png HTTP/1.1" 404 1886
"GET /static/images/slide2.jpg HTTP/1.1" 304 0
"GET /static/images/slide3.jpg HTTP/1.1" 304 0
"GET /authentication/logout/ HTTP/1.1" 302 0
"GET /landing_page/ HTTP/1.1" 200 9417
"GET /static/images/dashboard.png HTTP/1.1" 404 1874
"GET /static/images/chart-preview.png HTTP/1.1" 404 1886
"GET / HTTP/1.1" 200 9417
"GET /static/images/dashboard.png HTTP/1.1" 404 1874
"GET /static/images/chart-preview.png HTTP/1.1" 404 1886
Not Found: /favicon.ico
"GET /favicon.ico HTTP/1.1" 404 3911
Watching for file changes with StatReloader
"GET / HTTP/1.1" 200 9417
"GET /static/images/slide1.jpg HTTP/1.1" 304 0
"GET /static/images/dashboard.png HTTP/1.1" 404 1874
"GET /static/images/chart-preview.png HTTP/1.1" 404 1886
Not Found: /favicon.ico
"GET /favicon.ico HTTP/1.1" 404 3911
"GET /authentication/login/ HTTP/1.1" 200 3861
"GET /static/images/logo.webp HTTP/1.1" 304 0
"POST /authentication/login/ HTTP/1.1" 302 0
"GET /technician/dashboard/ HTTP/1.1" 200 22317
"GET /technician/models/ HTTP/1.1" 200 14753
"GET /technician/analytics/ HTTP/1.1" 200 16399
"GET /technician/soil-readings/ HTTP/1.1" 200 13051
"GET /technician/models/ HTTP/1.1" 200 14753
"GET /technician/reports/ HTTP/1.1" 200 15582
"GET /technician/analytics/ HTTP/1.1" 200 16399
"GET /technician/reports/ HTTP/1.1" 200 15582
"GET /technician/profile/ HTTP/1.1" 200 19352
"GET /media/users/profile_images/bruno.png HTTP/1.1" 304 0
"GET /technician/farm-locations/ HTTP/1.1" 200 15139
"GET /technician/farm/5/ HTTP/1.1" 200 15820
"GET /technician/soil-readings/?farm=5 HTTP/1.1" 200 12110
"GET /technician/soil-readings/ HTTP/1.1" 200 13051
"GET /technician/soil-readings/?farm=&date_from=&date_to=&status=Normal HTTP/1.1" 200 12134
"GET /technician/soil-readings/?farm=&date_from=&date_to=2025-07-14&status=Normal HTTP/1.1" 200 12144
"GET /technician/soil-readings/?farm=&date_from=&date_to=2025-07-14&status= HTTP/1.1" 200 13079
"GET /technician/soil-readings/?farm=4&date_from=&date_to=2025-07-14&status= HTTP/1.1" 200 12144
"GET /technician/soil-readings/?farm=&date_from=&date_to=2025-07-14&status= HTTP/1.1" 200 13079
"GET /technician/dashboard/ HTTP/1.1" 200 22317
"GET /technician/soil-readings/ HTTP/1.1" 200 13051
"GET /technician/soil-readings/ HTTP/1.1" 200 13051
"GET /technician/models/ HTTP/1.1" 200 14753
Watching for file changes with StatReloader
"GET / HTTP/1.1" 200 9417
"GET /static/images/slide1.jpg HTTP/1.1" 304 0
"GET /static/images/dashboard.png HTTP/1.1" 404 1874
"GET /static/images/chart-preview.png HTTP/1.1" 404 1886
Not Found: /favicon.ico
"GET /favicon.ico HTTP/1.1" 404 3911
"GET /static/images/slide2.jpg HTTP/1.1" 304 0
"GET /static/images/slide3.jpg HTTP/1.1" 304 0
"GET /authentication/login/ HTTP/1.1" 200 3861
"GET /static/images/logo.webp HTTP/1.1" 304 0
Not Found: /dashboard/
"GET /dashboard/ HTTP/1.1" 404 3908
"GET / HTTP/1.1" 200 9417
"GET /static/images/dashboard.png HTTP/1.1" 404 1874
"GET /static/images/chart-preview.png HTTP/1.1" 404 1886
"GET /authentication/signup/ HTTP/1.1" 200 5130
Not Found: /dashboard/
"GET /dashboard/ HTTP/1.1" 404 3908
C:\Users\user\Desktop\soils-sense\apps\technician\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\user\Desktop\soils-sense\ml\model_trainer.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\user\Desktop\soils-sense\apps\technician\urls.py changed, reloading.
Watching for file changes with StatReloader
"GET /authentication/signup/ HTTP/1.1" 200 5130
"GET /authentication/signup/ HTTP/1.1" 200 5130
"GET /authentication/signup/ HTTP/1.1" 200 5130
"GET /authentication/signup/ HTTP/1.1" 200 5130
"GET /authentication/signup/ HTTP/1.1" 200 5130
"GET /authentication/signup/ HTTP/1.1" 200 5130
Not Found: /dashboard/
"GET /dashboard/ HTTP/1.1" 404 3908
Not Found: /dashboard/
"GET /dashboard/ HTTP/1.1" 404 3908
"GET / HTTP/1.1" 200 9417
"GET /static/images/dashboard.png HTTP/1.1" 404 1874
"GET /static/images/chart-preview.png HTTP/1.1" 404 1886
"GET /authentication/signup/ HTTP/1.1" 200 5120
"GET /authentication/login/ HTTP/1.1" 200 3851
"GET /authentication/signup/ HTTP/1.1" 200 5120
"GET /authentication/login/ HTTP/1.1" 200 3851
"GET /authentication/reset/ HTTP/1.1" 200 3244
"GET /authentication/reset/ HTTP/1.1" 200 3136
"GET /authentication/reset/ HTTP/1.1" 200 3136
"GET /authentication/login/ HTTP/1.1" 200 3743
"POST /authentication/login/ HTTP/1.1" 302 0
Internal Server Error: /technician/dashboard/
Traceback (most recent call last):
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\apps\technician\views.py", line 134, in dashboard
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\backends\django.py", line 107, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\defaulttags.py", line 480, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\urls\base.py", line 98, in reverse
    resolved_url = resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\urls\resolvers.py", line 831, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'sensor_list' not found. 'sensor_list' is not a valid view function or pattern name.
"GET /technician/dashboard/ HTTP/1.1" 500 173200
Internal Server Error: /technician/dashboard/
Traceback (most recent call last):
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\apps\technician\views.py", line 134, in dashboard
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\backends\django.py", line 107, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\defaulttags.py", line 480, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\urls\base.py", line 98, in reverse
    resolved_url = resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\urls\resolvers.py", line 831, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'sensor_list' not found. 'sensor_list' is not a valid view function or pattern name.
"GET /technician/dashboard/ HTTP/1.1" 500 173203
Internal Server Error: /technician/dashboard/
Traceback (most recent call last):
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\apps\technician\views.py", line 134, in dashboard
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\backends\django.py", line 107, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\defaulttags.py", line 480, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\urls\base.py", line 98, in reverse
    resolved_url = resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\urls\resolvers.py", line 831, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'sensor_list' not found. 'sensor_list' is not a valid view function or pattern name.
"GET /technician/dashboard/ HTTP/1.1" 500 173200
Internal Server Error: /technician/dashboard/
Traceback (most recent call last):
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\apps\technician\views.py", line 134, in dashboard
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\backends\django.py", line 107, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\defaulttags.py", line 480, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\urls\base.py", line 98, in reverse
    resolved_url = resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\urls\resolvers.py", line 831, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'sensor_list' not found. 'sensor_list' is not a valid view function or pattern name.
"GET /technician/dashboard/ HTTP/1.1" 500 173203
Internal Server Error: /technician/dashboard/
Traceback (most recent call last):
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\apps\technician\views.py", line 134, in dashboard
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\backends\django.py", line 107, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\defaulttags.py", line 480, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\urls\base.py", line 98, in reverse
    resolved_url = resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\urls\resolvers.py", line 831, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'sensor_list' not found. 'sensor_list' is not a valid view function or pattern name.
"GET /technician/dashboard/ HTTP/1.1" 500 173203
Internal Server Error: /technician/dashboard/
Traceback (most recent call last):
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\apps\technician\views.py", line 134, in dashboard
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\backends\django.py", line 107, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\defaulttags.py", line 480, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\urls\base.py", line 98, in reverse
    resolved_url = resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\urls\resolvers.py", line 831, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'sensor_list' not found. 'sensor_list' is not a valid view function or pattern name.
"GET /technician/dashboard/ HTTP/1.1" 500 173200
Internal Server Error: /technician/dashboard/
Traceback (most recent call last):
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\apps\technician\views.py", line 134, in dashboard
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\backends\django.py", line 107, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\defaulttags.py", line 480, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\urls\base.py", line 98, in reverse
    resolved_url = resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\urls\resolvers.py", line 831, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'sensor_list' not found. 'sensor_list' is not a valid view function or pattern name.
"GET /technician/dashboard/ HTTP/1.1" 500 173203
Forbidden (CSRF token from POST incorrect.): /authentication/login/
"POST /authentication/login/ HTTP/1.1" 403 2503
Watching for file changes with StatReloader
"GET / HTTP/1.1" 200 9417
"GET /static/images/dashboard.png HTTP/1.1" 404 1987
"GET /static/images/chart-preview.png HTTP/1.1" 404 1999
"GET /authentication/login/ HTTP/1.1" 200 3743
"POST /authentication/login/ HTTP/1.1" 302 0
"GET /technician/dashboard/ HTTP/1.1" 200 22449
"GET /technician/sensors/ HTTP/1.1" 200 9472
"GET /technician/sensors/add/ HTTP/1.1" 200 10134
"POST /technician/sensors/add/ HTTP/1.1" 302 0
"GET /technician/sensors/ HTTP/1.1" 200 9768
"GET /technician/soil-readings/ HTTP/1.1" 200 13183
"GET /technician/farm-locations/ HTTP/1.1" 200 15271
"GET /technician/soil-readings/ HTTP/1.1" 200 13183
"GET /technician/sensors/ HTTP/1.1" 200 9768
"GET / HTTP/1.1" 200 9417
"GET /static/images/dashboard.png HTTP/1.1" 404 1987
"GET /static/images/chart-preview.png HTTP/1.1" 404 1999
"GET /authentication/login/ HTTP/1.1" 200 4021
"POST /authentication/login/ HTTP/1.1" 302 0
"GET /farmer/dashboard/ HTTP/1.1" 200 6956
"GET /farmer/soil-data/ HTTP/1.1" 200 29100
"GET /farmer/farm-management/ HTTP/1.1" 200 27690
"GET /farmer/dashboard/ HTTP/1.1" 200 6956
"GET /farmer/soil-data/ HTTP/1.1" 200 29100
"GET /farmer/predictions/ HTTP/1.1" 200 16754
C:\Users\user\Desktop\soils-sense\apps\farmer\models.py changed, reloading.
Watching for file changes with StatReloader
"GET /farmer/soil-data/ HTTP/1.1" 200 29100
"GET /farmer/soil-data/ HTTP/1.1" 200 29100
"GET /farmer/farm-management/ HTTP/1.1" 200 27690
"POST /farmer/add-farm/ HTTP/1.1" 302 0
"GET /farmer/farm-management/ HTTP/1.1" 200 29097
"GET /farmer/soil-data/ HTTP/1.1" 200 29505
"GET /technician/sensors/add/ HTTP/1.1" 302 0
"GET /authentication/login/?next=/technician/sensors/add/ HTTP/1.1" 200 3743
"POST /authentication/login/ HTTP/1.1" 302 0
"GET /technician/dashboard/ HTTP/1.1" 200 24264
"GET /technician/sensors/ HTTP/1.1" 200 9768
"GET /technician/sensors/add/ HTTP/1.1" 200 10178
"POST /technician/sensors/add/ HTTP/1.1" 302 0
"GET /technician/sensors/ HTTP/1.1" 200 10109
"GET /farmer/dashboard/ HTTP/1.1" 200 13494
"GET /farmer/soil-data/ HTTP/1.1" 200 41879
"GET /farmer/soil-data/ HTTP/1.1" 200 43230
"GET /farmer/soil-data/ HTTP/1.1" 200 43230
"GET /farmer/recommendations/ HTTP/1.1" 200 11891
"GET /farmer/soil-data/ HTTP/1.1" 200 52700
"GET /farmer/recommendations/ HTTP/1.1" 200 11891
"GET /farmer/analytics/ HTTP/1.1" 200 19152
"GET /farmer/predictions/ HTTP/1.1" 200 20001
"GET /farmer/soil-data/ HTTP/1.1" 200 54051
"GET /farmer/recommendations/ HTTP/1.1" 200 11784
"GET /farmer/profile/ HTTP/1.1" 200 12377
"GET /authentication/logout/ HTTP/1.1" 302 0
"GET /landing_page/ HTTP/1.1" 200 9417
"GET /static/images/dashboard.png HTTP/1.1" 404 1987
"GET /static/images/chart-preview.png HTTP/1.1" 404 1999
"GET /authentication/login/ HTTP/1.1" 200 3743
"POST /authentication/login/ HTTP/1.1" 302 0
"GET /farmer/dashboard/ HTTP/1.1" 200 12669
"GET /farmer/soil-data/ HTTP/1.1" 200 48223
"GET /authentication/signup/ HTTP/1.1" 200 5012
"GET /admin/ HTTP/1.1" 302 0
"GET /admin/login/?next=/admin/ HTTP/1.1" 200 3396
"GET /static/vendor/adminlte/css/adminlte.min.css HTTP/1.1" 304 0
"GET /static/jazzmin/css/main.css HTTP/1.1" 304 0
"GET /static/vendor/fontawesome-free/css/all.min.css HTTP/1.1" 304 0
"GET /static/vendor/bootstrap/js/bootstrap.min.js HTTP/1.1" 304 0
"GET /static/vendor/adminlte/img/AdminLTELogo.png HTTP/1.1" 304 0
"GET /static/admin/js/vendor/jquery/jquery.js HTTP/1.1" 304 0
"GET /static/vendor/adminlte/js/adminlte.min.js HTTP/1.1" 304 0
"GET /static/vendor/fontawesome-free/webfonts/fa-solid-900.woff2 HTTP/1.1" 304 0
Forbidden: /administrator/
"GET /administrator/ HTTP/1.1" 403 47
"GET /authentication/logout/ HTTP/1.1" 302 0
"GET /landing_page/ HTTP/1.1" 200 9417
"GET /static/images/chart-preview.png HTTP/1.1" 404 1999
"GET /static/images/dashboard.png HTTP/1.1" 404 1987
"GET /authentication/login/ HTTP/1.1" 200 3743
"POST /authentication/login/ HTTP/1.1" 302 0
"GET /administrator/ HTTP/1.1" 200 6479
"GET /administrator/ml-models/ HTTP/1.1" 200 25729
"GET /administrator/ml-models/irrigation_recommendation/?algorithm=random_forest HTTP/1.1" 200 21709
"GET /administrator/data/ HTTP/1.1" 200 6322
"GET /administrator/reports/ HTTP/1.1" 200 6129
Watching for file changes with StatReloader
"GET /technician/ HTTP/1.1" 302 0
"GET /technician/dashboard/ HTTP/1.1" 200 25371
"GET /technician/soil-readings/ HTTP/1.1" 302 0
"GET /authentication/login/?next=/technician/soil-readings/ HTTP/1.1" 200 3743
"POST /authentication/login/ HTTP/1.1" 302 0
"GET /farmer/dashboard/ HTTP/1.1" 200 12669
"GET /technician/farm-locations/ HTTP/1.1" 200 16337
"GET /technician/soil-readings/ HTTP/1.1" 302 0
"GET /authentication/login/?next=/technician/soil-readings/ HTTP/1.1" 200 3743
"GET /technician/soil-readings/ HTTP/1.1" 302 0
"GET /authentication/login/?next=/technician/soil-readings/ HTTP/1.1" 200 3743
"GET /technician/farm-locations/ HTTP/1.1" 200 16337
"GET /technician/soil-readings/ HTTP/1.1" 302 0
"GET /authentication/login/?next=/technician/soil-readings/ HTTP/1.1" 200 3743
"GET /technician/reports/ HTTP/1.1" 200 15714
"GET /technician/analytics/ HTTP/1.1" 200 17359
"GET /technician/profile/ HTTP/1.1" 200 19559
"GET /technician/reports/ HTTP/1.1" 200 15714
"GET /technician/analytics/ HTTP/1.1" 200 17359
"GET /technician/soil-readings/ HTTP/1.1" 302 0
"GET /authentication/login/?next=/technician/soil-readings/ HTTP/1.1" 200 3743
"GET /technician/dashboard/ HTTP/1.1" 200 25371
"GET /technician/farm-locations/ HTTP/1.1" 200 16337
"GET /technician/models/ HTTP/1.1" 200 14885
C:\Users\user\Desktop\soils-sense\apps\technician\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\user\Desktop\soils-sense\apps\technician\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\user\Desktop\soils-sense\apps\technician\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\user\Desktop\soils-sense\apps\technician\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\user\Desktop\soils-sense\apps\technician\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\user\Desktop\soils-sense\apps\technician\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\user\Desktop\soils-sense\apps\technician\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\user\Desktop\soils-sense\apps\technician\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\user\Desktop\soils-sense\apps\technician\views.py changed, reloading.
Watching for file changes with StatReloader
"GET /authentication/logout/ HTTP/1.1" 302 0
"GET /landing_page/ HTTP/1.1" 200 9417
"GET /static/images/dashboard.png HTTP/1.1" 404 1987
"GET /static/images/chart-preview.png HTTP/1.1" 404 1999
"GET /technician/ HTTP/1.1" 302 0
"GET /technician/dashboard/ HTTP/1.1" 302 0
"GET /authentication/login/?next=/technician/dashboard/ HTTP/1.1" 200 3743
"GET /authentication/login/?next=/technician/reports/ HTTP/1.1" 200 3743
"POST /authentication/login/ HTTP/1.1" 302 0
"GET /technician/dashboard/ HTTP/1.1" 200 25416
"GET /technician/farm-locations/ HTTP/1.1" 200 16337
"GET /technician/soil-readings/ HTTP/1.1" 200 35676
"GET /technician/soil-readings/?page=2 HTTP/1.1" 200 35828
"GET /technician/soil-readings/ HTTP/1.1" 200 35676
"GET /technician/analytics/ HTTP/1.1" 200 17359
"GET /authentication/logout/ HTTP/1.1" 302 0
"GET /landing_page/ HTTP/1.1" 200 9417
"GET /static/images/dashboard.png HTTP/1.1" 404 1987
"GET /static/images/chart-preview.png HTTP/1.1" 404 1999
Not Found: /privacy
"GET /privacy HTTP/1.1" 404 4069
Not Found: /terms
"GET /terms HTTP/1.1" 404 4063
"GET /technician/ HTTP/1.1" 302 0
"GET /technician/dashboard/ HTTP/1.1" 302 0
"GET /authentication/login/?next=/technician/dashboard/ HTTP/1.1" 200 3743
Watching for file changes with StatReloader
"GET / HTTP/1.1" 200 9821
"GET /static/images/slide1.jpg HTTP/1.1" 304 0
"GET /static/images/dashboard.png HTTP/1.1" 404 1987
"GET /static/images/chart-preview.png HTTP/1.1" 404 1999
Not Found: /favicon.ico
"GET /favicon.ico HTTP/1.1" 404 4081
"GET /static/images/wheat.jpg HTTP/1.1" 200 1094218
"GET /static/images/lady.jpg HTTP/1.1" 200 1017834
"GET /static/images/lady3.jpg HTTP/1.1" 200 859813
"GET /static/images/ladyone.jpg HTTP/1.1" 200 911834
"GET /authentication/signup/ HTTP/1.1" 200 4168
"GET /static/images/logo.webp HTTP/1.1" 304 0
"GET /authentication/login/ HTTP/1.1" 200 2899
Internal Server Error: /authentication/login/
Traceback (most recent call last):
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\db\backends\utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\db\backends\sqlite3\base.py", line 360, in execute
    return super().execute(query, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
sqlite3.OperationalError: no such column: authentication_user.reset_code

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\views\generic\base.py", line 105, in view
    return self.dispatch(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\views\generic\base.py", line 144, in dispatch
    return handler(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\authentication\views.py", line 40, in post
    user = authenticate(request, username=username, password=password)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\views\decorators\debug.py", line 75, in sensitive_variables_wrapper
    return func(*func_args, **func_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\contrib\auth\__init__.py", line 114, in authenticate
    user = backend.authenticate(request, **credentials)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\contrib\auth\backends.py", line 65, in authenticate
    user = UserModel._default_manager.get_by_natural_key(username)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\contrib\auth\base_user.py", line 37, in get_by_natural_key
    return self.get(**{self.model.USERNAME_FIELD: username})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\db\models\manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\db\models\query.py", line 629, in get
    num = len(clone)
          ^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\db\models\query.py", line 366, in __len__
    self._fetch_all()
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\db\models\query.py", line 1949, in _fetch_all
    self._result_cache = list(self._iterable_class(self))
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\db\models\query.py", line 91, in __iter__
    results = compiler.execute_sql(
              ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\db\models\sql\compiler.py", line 1623, in execute_sql
    cursor.execute(sql, params)
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\db\backends\utils.py", line 122, in execute
    return super().execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\db\backends\utils.py", line 79, in execute
    return self._execute_with_wrappers(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\db\backends\utils.py", line 92, in _execute_with_wrappers
    return executor(sql, params, many, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\db\backends\utils.py", line 100, in _execute
    with self.db.wrap_database_errors:
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\db\utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\db\backends\utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\db\backends\sqlite3\base.py", line 360, in execute
    return super().execute(query, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
django.db.utils.OperationalError: no such column: authentication_user.reset_code
"POST /authentication/login/ HTTP/1.1" 500 177745
"POST /authentication/login/ HTTP/1.1" 302 0
"GET /technician/dashboard/ HTTP/1.1" 200 25426
"GET /technician/farm-locations/ HTTP/1.1" 200 16337
"GET /authentication/logout/ HTTP/1.1" 302 0
"GET /landing_page/ HTTP/1.1" 200 9821
"GET /static/images/dashboard.png HTTP/1.1" 404 1987
"GET /static/images/chart-preview.png HTTP/1.1" 404 1999
"GET /authentication/login/ HTTP/1.1" 200 2899
"GET /authentication/reset/ HTTP/1.1" 200 2292
"POST /authentication/reset/ HTTP/1.1" 302 0
"GET /authentication/reset/done/ HTTP/1.1" 200 1384
"GET /authentication/login/ HTTP/1.1" 200 2899
"GET /authentication/reset/ HTTP/1.1" 200 2292
"POST /authentication/reset/ HTTP/1.1" 302 0
"GET /authentication/reset/done/ HTTP/1.1" 200 1384
"GET /authentication/reset/MQ/ct3dl1-dea545994be1496f341eac36d70f0905/ HTTP/1.1" 302 0
"GET /authentication/reset/MQ/set-password/ HTTP/1.1" 200 3090
"GET /authentication/login/ HTTP/1.1" 200 2899
"POST /authentication/login/ HTTP/1.1" 302 0
"GET /technician/dashboard/ HTTP/1.1" 200 25426
"GET /authentication/logout/ HTTP/1.1" 302 0
"GET /landing_page/ HTTP/1.1" 200 9821
"GET /static/images/dashboard.png HTTP/1.1" 404 1987
"GET /static/images/chart-preview.png HTTP/1.1" 404 1999
"GET /authentication/login/ HTTP/1.1" 200 2899
"POST /authentication/login/ HTTP/1.1" 302 0
"GET /farmer/dashboard/ HTTP/1.1" 200 15555
Internal Server Error: /farmer/notifications/unread-count/
Traceback (most recent call last):
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\db\backends\utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\db\backends\sqlite3\base.py", line 360, in execute
    return super().execute(query, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
sqlite3.OperationalError: no such table: farmer_notification

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 59, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\apps\farmer\utils.py", line 19, in _wrapped_view
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\apps\farmer\views.py", line 1200, in get_unread_count
    count = Notification.objects.filter(user=request.user, is_read=False).count()
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\db\models\query.py", line 604, in count
    return self.query.get_count(using=self.db)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\db\models\sql\query.py", line 644, in get_count
    return obj.get_aggregation(using, {"__count": Count("*")})["__count"]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\db\models\sql\query.py", line 626, in get_aggregation
    result = compiler.execute_sql(SINGLE)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\db\models\sql\compiler.py", line 1623, in execute_sql
    cursor.execute(sql, params)
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\db\backends\utils.py", line 122, in execute
    return super().execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\db\backends\utils.py", line 79, in execute
    return self._execute_with_wrappers(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\db\backends\utils.py", line 92, in _execute_with_wrappers
    return executor(sql, params, many, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\db\backends\utils.py", line 100, in _execute
    with self.db.wrap_database_errors:
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\db\utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\db\backends\utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\db\backends\sqlite3\base.py", line 360, in execute
    return super().execute(query, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
django.db.utils.OperationalError: no such table: farmer_notification
"GET /farmer/notifications/unread-count/ HTTP/1.1" 500 144166
"GET /farmer/profile/ HTTP/1.1" 200 15292
Internal Server Error: /farmer/notifications/unread-count/
Traceback (most recent call last):
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\db\backends\utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\db\backends\sqlite3\base.py", line 360, in execute
    return super().execute(query, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
sqlite3.OperationalError: no such table: farmer_notification

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 59, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\apps\farmer\utils.py", line 19, in _wrapped_view
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\apps\farmer\views.py", line 1200, in get_unread_count
    count = Notification.objects.filter(user=request.user, is_read=False).count()
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\db\models\query.py", line 604, in count
    return self.query.get_count(using=self.db)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\db\models\sql\query.py", line 644, in get_count
    return obj.get_aggregation(using, {"__count": Count("*")})["__count"]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\db\models\sql\query.py", line 626, in get_aggregation
    result = compiler.execute_sql(SINGLE)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\db\models\sql\compiler.py", line 1623, in execute_sql
    cursor.execute(sql, params)
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\db\backends\utils.py", line 122, in execute
    return super().execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\db\backends\utils.py", line 79, in execute
    return self._execute_with_wrappers(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\db\backends\utils.py", line 92, in _execute_with_wrappers
    return executor(sql, params, many, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\db\backends\utils.py", line 100, in _execute
    with self.db.wrap_database_errors:
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\db\utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\db\backends\utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\db\backends\sqlite3\base.py", line 360, in execute
    return super().execute(query, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
django.db.utils.OperationalError: no such table: farmer_notification
"GET /farmer/notifications/unread-count/ HTTP/1.1" 500 144164
Internal Server Error: /farmer/notifications/
Traceback (most recent call last):
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\db\backends\utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\db\backends\sqlite3\base.py", line 360, in execute
    return super().execute(query, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
sqlite3.OperationalError: no such table: farmer_notification

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 59, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\apps\farmer\utils.py", line 19, in _wrapped_view
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\apps\farmer\views.py", line 1187, in notifications
    return render(request, 'farmer/notifications.html', {
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\backends\django.py", line 107, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loader_tags.py", line 65, in render
    result = block.nodelist.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\defaulttags.py", line 326, in render
    if match:
       ^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\db\models\query.py", line 398, in __bool__
    self._fetch_all()
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\db\models\query.py", line 1949, in _fetch_all
    self._result_cache = list(self._iterable_class(self))
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\db\models\query.py", line 91, in __iter__
    results = compiler.execute_sql(
              ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\db\models\sql\compiler.py", line 1623, in execute_sql
    cursor.execute(sql, params)
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\db\backends\utils.py", line 122, in execute
    return super().execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\db\backends\utils.py", line 79, in execute
    return self._execute_with_wrappers(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\db\backends\utils.py", line 92, in _execute_with_wrappers
    return executor(sql, params, many, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\db\backends\utils.py", line 100, in _execute
    with self.db.wrap_database_errors:
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\db\utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\db\backends\utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\db\backends\sqlite3\base.py", line 360, in execute
    return super().execute(query, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
django.db.utils.OperationalError: no such table: farmer_notification
"GET /farmer/notifications/ HTTP/1.1" 500 217375
"GET /farmer/notifications/ HTTP/1.1" 200 9636
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/recommendations/ HTTP/1.1" 200 14567
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/analytics/ HTTP/1.1" 200 24105
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/predictions/ HTTP/1.1" 200 19781
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/get-latest-readings/6/ HTTP/1.1" 200 206
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"POST /farmer/predictions/ HTTP/1.1" 302 0
"GET /farmer/predictions/ HTTP/1.1" 200 23340
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/analytics/ HTTP/1.1" 200 24105
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/notifications/ HTTP/1.1" 200 12036
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/predictions/ HTTP/1.1" 200 22815
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/get-latest-readings/6/ HTTP/1.1" 200 206
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"POST /farmer/predictions/ HTTP/1.1" 302 0
"GET /farmer/predictions/ HTTP/1.1" 200 26533
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/notifications/ HTTP/1.1" 200 14645
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/farm-management/ HTTP/1.1" 200 31693
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/dashboard/ HTTP/1.1" 200 15555
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/farm-management/ HTTP/1.1" 200 31693
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/soil-data/ HTTP/1.1" 200 72363
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/dashboard/ HTTP/1.1" 200 15555
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET / HTTP/1.1" 200 9821
"GET /static/images/chart-preview.png HTTP/1.1" 404 1999
"GET /static/images/dashboard.png HTTP/1.1" 404 1987
"GET /authentication/login/ HTTP/1.1" 200 2899
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"POST /authentication/login/ HTTP/1.1" 302 0
"GET /technician/dashboard/ HTTP/1.1" 200 25422
"GET /technician/farm-locations/ HTTP/1.1" 200 16337
"GET /technician/soil-readings/ HTTP/1.1" 200 35676
"GET /technician/sensors/ HTTP/1.1" 200 10109
"GET /technician/models/ HTTP/1.1" 200 14885
Forbidden: /farmer/notifications/unread-count/
"GET /farmer/notifications/unread-count/ HTTP/1.1" 403 47
Forbidden: /farmer/notifications/unread-count/
"GET /farmer/notifications/unread-count/ HTTP/1.1" 403 47
Forbidden: /farmer/notifications/unread-count/
"GET /farmer/notifications/unread-count/ HTTP/1.1" 403 47
Forbidden: /farmer/notifications/unread-count/
"GET /farmer/notifications/unread-count/ HTTP/1.1" 403 47
Forbidden: /farmer/notifications/unread-count/
"GET /farmer/notifications/unread-count/ HTTP/1.1" 403 47
Forbidden: /farmer/notifications/unread-count/
"GET /farmer/notifications/unread-count/ HTTP/1.1" 403 47
Forbidden: /farmer/notifications/unread-count/
"GET /farmer/notifications/unread-count/ HTTP/1.1" 403 47
"GET /technician/models/ HTTP/1.1" 200 13817
Forbidden: /farmer/notifications/unread-count/
"GET /farmer/notifications/unread-count/ HTTP/1.1" 403 47
Forbidden: /farmer/notifications/unread-count/
"GET /farmer/notifications/unread-count/ HTTP/1.1" 403 47
"GET /technician/models/ HTTP/1.1" 200 15015
Forbidden: /farmer/notifications/unread-count/
"GET /farmer/notifications/unread-count/ HTTP/1.1" 403 47
"GET /technician/models/ HTTP/1.1" 200 15150
Forbidden: /farmer/notifications/unread-count/
"GET /farmer/notifications/unread-count/ HTTP/1.1" 403 47
"GET /technician/models/ HTTP/1.1" 200 15150
Forbidden: /farmer/notifications/unread-count/
"GET /farmer/notifications/unread-count/ HTTP/1.1" 403 47
"GET /technician/dashboard/ HTTP/1.1" 200 25688
"GET /technician/farm-locations/ HTTP/1.1" 200 16602
Forbidden: /farmer/notifications/unread-count/
"GET /farmer/notifications/unread-count/ HTTP/1.1" 403 47
"GET /technician/sensors/ HTTP/1.1" 200 10374
"GET /technician/profile/ HTTP/1.1" 200 19749
"GET /media/users/profile_images/bruno.png HTTP/1.1" 304 0
"GET /technician/settings/ HTTP/1.1" 200 17286
"GET /technician/profile/ HTTP/1.1" 200 19749
- Broken pipe from ('127.0.0.1', 61110)
"GET /technician/models/ HTTP/1.1" 200 15150
Forbidden: /farmer/notifications/unread-count/
"GET /farmer/notifications/unread-count/ HTTP/1.1" 403 47
C:\Users\user\Desktop\soils-sense\apps\farmer\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\user\Desktop\soils-sense\apps\farmer\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\user\Desktop\soils-sense\apps\farmer\views.py changed, reloading.
Watching for file changes with StatReloader
Forbidden: /farmer/notifications/unread-count/
"GET /farmer/notifications/unread-count/ HTTP/1.1" 403 47
Forbidden: /farmer/notifications/unread-count/
"GET /farmer/notifications/unread-count/ HTTP/1.1" 403 47
Forbidden: /farmer/soil-data/
"GET /farmer/soil-data/ HTTP/1.1" 403 47
Not Found: /farmer
"GET /farmer HTTP/1.1" 404 4066
Not Found: /farmer/login
"GET /farmer/login HTTP/1.1" 404 10871
"GET /authentication/login HTTP/1.1" 301 0
"GET /authentication/login/ HTTP/1.1" 200 2899
"POST /authentication/login/ HTTP/1.1" 302 0
- Broken pipe from ('127.0.0.1', 61417)
Watching for file changes with StatReloader
"GET / HTTP/1.1" 200 15668
"GET /static/images/dashboard.png HTTP/1.1" 404 1987
"GET /static/images/chart-preview.png HTTP/1.1" 404 1999
"GET /static/images/ladyone.jpg HTTP/1.1" 304 0
"GET /static/images/garden2.jpg HTTP/1.1" 200 4417134
"GET /static/images/lady.jpg HTTP/1.1" 304 0
"GET /static/images/lady3.jpg HTTP/1.1" 304 0
"GET /authentication/login/ HTTP/1.1" 200 2899
"POST /authentication/login/ HTTP/1.1" 302 0
"GET /farmer/dashboard/ HTTP/1.1" 200 25639
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/farm-management/ HTTP/1.1" 200 31693
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/soil-data/ HTTP/1.1" 200 72363
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/notifications/ HTTP/1.1" 200 14645
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/predictions/ HTTP/1.1" 200 26014
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /authentication/logout/ HTTP/1.1" 302 0
"GET /landing_page/ HTTP/1.1" 200 15668
"GET /static/images/dashboard.png HTTP/1.1" 404 1987
"GET /static/images/chart-preview.png HTTP/1.1" 404 1999
"GET /static/images/ladyone.jpg HTTP/1.1" 304 0
"GET /authentication/login/ HTTP/1.1" 200 2899
"POST /authentication/login/ HTTP/1.1" 302 0
Internal Server Error: /technician/dashboard/
Traceback (most recent call last):
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\base.py", line 204, in _get_response
    self.check_response(response, callback)
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\base.py", line 332, in check_response
    raise ValueError(
ValueError: The view apps.technician.views.dashboard didn't return an HttpResponse object. It returned None instead.
"GET /technician/dashboard/ HTTP/1.1" 500 71702
"GET /farmer/dashboard/ HTTP/1.1" 200 25639
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
Internal Server Error: /technician/dashboard/
Traceback (most recent call last):
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\base.py", line 204, in _get_response
    self.check_response(response, callback)
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\base.py", line 332, in check_response
    raise ValueError(
ValueError: The view apps.technician.views.dashboard didn't return an HttpResponse object. It returned None instead.
"GET /technician/dashboard/ HTTP/1.1" 500 71839
Internal Server Error: /technician/dashboard/
Traceback (most recent call last):
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\base.py", line 204, in _get_response
    self.check_response(response, callback)
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\base.py", line 332, in check_response
    raise ValueError(
ValueError: The view apps.technician.views.dashboard didn't return an HttpResponse object. It returned None instead.
"GET /technician/dashboard/ HTTP/1.1" 500 71839
"GET /authentication/logout/ HTTP/1.1" 302 0
"GET /landing_page/ HTTP/1.1" 200 15668
"GET /static/images/dashboard.png HTTP/1.1" 404 1987
"GET /static/images/chart-preview.png HTTP/1.1" 404 1999
"GET /authentication/login/ HTTP/1.1" 200 2899
"POST /authentication/login/ HTTP/1.1" 302 0
"GET /technician/dashboard/ HTTP/1.1" 200 25691
"GET /technician/sensors/ HTTP/1.1" 200 10374
"GET /technician/sensors/1/edit/ HTTP/1.1" 200 10480
"GET /authentication/logout/ HTTP/1.1" 302 0
"GET /landing_page/ HTTP/1.1" 200 15668
"GET /static/images/dashboard.png HTTP/1.1" 404 1987
"GET /static/images/chart-preview.png HTTP/1.1" 404 1999
"GET /static/images/ladyone.jpg HTTP/1.1" 304 0
"GET /authentication/login/ HTTP/1.1" 200 2899
"POST /authentication/login/ HTTP/1.1" 302 0
"GET /farmer/dashboard/ HTTP/1.1" 200 25639
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/notifications/ HTTP/1.1" 200 14645
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/notifications/ HTTP/1.1" 200 14645
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/dashboard/ HTTP/1.1" 200 25639
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/notifications/ HTTP/1.1" 200 14645
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/dashboard/ HTTP/1.1" 200 25639
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /authentication/logout/ HTTP/1.1" 302 0
"GET /landing_page/ HTTP/1.1" 200 15668
"GET /static/images/dashboard.png HTTP/1.1" 404 1987
"GET /static/images/chart-preview.png HTTP/1.1" 404 1999
"GET /authentication/login/ HTTP/1.1" 200 2899
"POST /authentication/login/ HTTP/1.1" 302 0
"GET /technician/dashboard/ HTTP/1.1" 200 25691
"GET /technician/sensors/ HTTP/1.1" 200 10374
"GET /authentication/logout/ HTTP/1.1" 302 0
"GET /landing_page/ HTTP/1.1" 200 15668
"GET /static/images/dashboard.png HTTP/1.1" 404 1987
"GET /static/images/chart-preview.png HTTP/1.1" 404 1999
"GET /authentication/login/ HTTP/1.1" 200 2899
"POST /authentication/login/ HTTP/1.1" 302 0
"GET /farmer/dashboard/ HTTP/1.1" 200 25636
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/notifications/ HTTP/1.1" 200 14645
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /authentication/logout/ HTTP/1.1" 302 0
"GET /landing_page/ HTTP/1.1" 200 15668
"GET /static/images/dashboard.png HTTP/1.1" 404 1987
"GET /static/images/chart-preview.png HTTP/1.1" 404 1999
"GET /static/images/ladyone.jpg HTTP/1.1" 304 0
"GET /authentication/login/ HTTP/1.1" 200 2899
- Broken pipe from ('127.0.0.1', 61868)
"POST /authentication/login/ HTTP/1.1" 302 0
"GET /technician/dashboard/ HTTP/1.1" 200 25691
Watching for file changes with StatReloader
Watching for file changes with StatReloader
Watching for file changes with StatReloader
Watching for file changes with StatReloader
"GET / HTTP/1.1" 200 15668
"GET /static/images/dashboard.png HTTP/1.1" 404 1987
"GET /static/images/chart-preview.png HTTP/1.1" 404 1999
"GET /static/images/ladyone.jpg HTTP/1.1" 304 0
Not Found: /favicon.ico
"GET /favicon.ico HTTP/1.1" 404 4081
"GET /authentication/login/ HTTP/1.1" 200 2899
"POST /authentication/login/ HTTP/1.1" 200 2913
"POST /authentication/login/ HTTP/1.1" 200 2913
"POST /authentication/login/ HTTP/1.1" 200 2913
"GET /authentication/signup/ HTTP/1.1" 200 4168
"POST /authentication/signup/ HTTP/1.1" 302 0
"GET /authentication/login/ HTTP/1.1" 200 3184
"GET /authentication/signup/ HTTP/1.1" 200 4168
"POST /authentication/signup/ HTTP/1.1" 302 0
"GET /authentication/login/ HTTP/1.1" 200 3184
"POST /authentication/login/ HTTP/1.1" 302 0
"GET /administrator/ HTTP/1.1" 200 6744
"GET /administrator/ HTTP/1.1" 200 6744
"GET /administrator/users/ HTTP/1.1" 200 33470
Not Found: /media/users/profiles/default.webp
"GET /media/users/profiles/default.webp HTTP/1.1" 404 4211
"GET /static/images/default.webp HTTP/1.1" 200 24145
Not Found: /media/users/profiles/default.webp
"GET /media/users/profiles/default.webp HTTP/1.1" 404 4211
"GET /administrator/ HTTP/1.1" 200 6744
"GET /administrator/users/ HTTP/1.1" 200 33483
Not Found: /media/users/profiles/default.webp
"GET /media/users/profiles/default.webp HTTP/1.1" 404 4211
Not Found: /media/users/profiles/default.webp
"GET /media/users/profiles/default.webp HTTP/1.1" 404 4211
"GET /administrator/data/ HTTP/1.1" 200 31474
Not Found: /media/users/profiles/default.webp
"GET /media/users/profiles/default.webp HTTP/1.1" 404 4211
Not Found: /media/users/profiles/default.webp
"GET /media/users/profiles/default.webp HTTP/1.1" 404 4211
"GET /administrator/notifications/ HTTP/1.1" 200 6641
"GET /administrator/notifications/ HTTP/1.1" 200 6641
"GET /administrator/sensors/ HTTP/1.1" 200 6600
"GET /administrator/reports/ HTTP/1.1" 200 6394
"GET /administrator/ml-models/ HTTP/1.1" 200 27010
"GET /administrator/ml-models/irrigation_recommendation/?algorithm=gradient_boosting HTTP/1.1" 200 27357
"GET / HTTP/1.1" 200 15668
"GET /static/images/dashboard.png HTTP/1.1" 404 1987
"GET /static/images/chart-preview.png HTTP/1.1" 404 1999
"GET /authentication/signup/ HTTP/1.1" 200 4168
"GET /authentication/login/ HTTP/1.1" 200 2899
C:\Users\user\Desktop\soils-sense\apps\technician\views.py changed, reloading.
Watching for file changes with StatReloader
"GET /authentication/login/ HTTP/1.1" 200 2899
C:\Users\user\Desktop\soils-sense\ml\__init__.py changed, reloading.
Watching for file changes with StatReloader
"POST /authentication/login/ HTTP/1.1" 302 0
"GET /administrator/ HTTP/1.1" 200 6744
"GET /authentication/logout/ HTTP/1.1" 302 0
"GET /landing_page/ HTTP/1.1" 200 15668
"GET /static/images/dashboard.png HTTP/1.1" 404 1987
"GET /static/images/chart-preview.png HTTP/1.1" 404 1999
"GET /authentication/login/ HTTP/1.1" 200 2899
"POST /authentication/login/ HTTP/1.1" 302 0
"GET /farmer/dashboard/ HTTP/1.1" 200 14273
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/farm-management/ HTTP/1.1" 200 30607
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"POST /farmer/add-farm/ HTTP/1.1" 302 0
"GET /farmer/farm-management/ HTTP/1.1" 200 32019
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"POST /farmer/add-crop/ HTTP/1.1" 302 0
"GET /farmer/farm-management/ HTTP/1.1" 200 33665
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/soil-data/ HTTP/1.1" 200 23163
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/farm-management/ HTTP/1.1" 200 33335
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/farm-management/ HTTP/1.1" 200 33335
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/soil-data/ HTTP/1.1" 200 23163
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/api/sensor-data-json/3/?minutes=30 HTTP/1.1" 200 33
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/api/sensor-data-json/3/?minutes=30 HTTP/1.1" 200 33
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/api/sensor-data-json/3/?minutes=30 HTTP/1.1" 200 33
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /authentication/logout/ HTTP/1.1" 302 0
"GET /landing_page/ HTTP/1.1" 200 15668
"GET /static/images/dashboard.png HTTP/1.1" 404 1987
"GET /static/images/chart-preview.png HTTP/1.1" 404 1999
"GET /authentication/login/ HTTP/1.1" 200 2899
"POST /authentication/login/ HTTP/1.1" 302 0
"GET /technician/dashboard/ HTTP/1.1" 200 25658
"GET /technician/sensors/ HTTP/1.1" 200 10356
"GET /technician/sensors/add/ HTTP/1.1" 200 10459
"POST /technician/sensors/add/ HTTP/1.1" 302 0
"GET /technician/sensors/ HTTP/1.1" 200 10688
"GET /technician/sensors/add/ HTTP/1.1" 200 10459
C:\Users\user\Desktop\soils-sense\ml\__init__.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\user\Desktop\soils-sense\apps\technician\views.py changed, reloading.
Watching for file changes with StatReloader
"GET /technician/dashboard/ HTTP/1.1" 200 25665
"GET /technician/models/ HTTP/1.1" 200 32541
C:\Users\user\Desktop\soils-sense\apps\technician\views.py changed, reloading.
Watching for file changes with StatReloader
"GET /technician/models/ HTTP/1.1" 200 16913
"GET /technician/models/ HTTP/1.1" 200 16913
C:\Users\user\Desktop\soils-sense\apps\technician\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\user\Desktop\soils-sense\apps\technician\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\user\Desktop\soils-sense\apps\technician\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\user\Desktop\soils-sense\apps\technician\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\user\Desktop\soils-sense\apps\technician\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\user\Desktop\soils-sense\apps\technician\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\user\Desktop\soils-sense\apps\technician\views.py changed, reloading.
Watching for file changes with StatReloader
"GET /technician/models/ HTTP/1.1" 200 15150
"GET /technician/models/ HTTP/1.1" 200 46938
"GET /technician/models/ HTTP/1.1" 200 37686
"GET /technician/sensors/ HTTP/1.1" 200 10688
"GET /technician/sensors/add/ HTTP/1.1" 200 10459
Internal Server Error: /technician/sensors/add/
Traceback (most recent call last):
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 59, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 59, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\apps\technician\views.py", line 56, in sensor_add
    return render(request, 'technician/sensor_form.html', {'form': form, 'title': 'Add Sensor'})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\backends\django.py", line 107, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loader_tags.py", line 65, in render
    result = block.nodelist.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loader_tags.py", line 65, in render
    result = block.nodelist.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\defaulttags.py", line 480, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\urls\base.py", line 98, in reverse
    resolved_url = resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\urls\resolvers.py", line 831, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'sensors' not found. 'sensors' is not a valid view function or pattern name.
"GET /technician/sensors/add/ HTTP/1.1" 500 190769
Internal Server Error: /technician/sensors/add/
Traceback (most recent call last):
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 59, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 59, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\apps\technician\views.py", line 56, in sensor_add
    return render(request, 'technician/sensor_form.html', {'form': form, 'title': 'Add Sensor'})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\backends\django.py", line 107, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loader_tags.py", line 65, in render
    result = block.nodelist.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\loader_tags.py", line 65, in render
    result = block.nodelist.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\template\defaulttags.py", line 480, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\urls\base.py", line 98, in reverse
    resolved_url = resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\Desktop\soils-sense\.venv\Lib\site-packages\django\urls\resolvers.py", line 831, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'sensors' not found. 'sensors' is not a valid view function or pattern name.
"GET /technician/sensors/add/ HTTP/1.1" 500 190632
C:\Users\user\Desktop\soils-sense\apps\technician\views.py changed, reloading.
Watching for file changes with StatReloader
"GET /technician/sensors/add/ HTTP/1.1" 200 15778
"GET /technician/sensors/ HTTP/1.1" 200 10688
"GET /technician/sensors/add/ HTTP/1.1" 200 15778
"GET /technician/sensors/add/ HTTP/1.1" 200 16207
"GET /technician/models/ HTTP/1.1" 200 37686
"GET /technician/sensors/ HTTP/1.1" 200 10688
"GET /technician/sensors/add/ HTTP/1.1" 200 15778
C:\Users\user\Desktop\soils-sense\apps\technician\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\user\Desktop\soils-sense\apps\technician\forms.py changed, reloading.
Watching for file changes with StatReloader
"GET /technician/sensors/add/ HTTP/1.1" 200 16113
C:\Users\user\Desktop\soils-sense\apps\technician\forms.py changed, reloading.
Watching for file changes with StatReloader
"GET /technician/sensors/add/ HTTP/1.1" 200 16170
"GET /technician/sensors/add/ HTTP/1.1" 200 16150
"GET /technician/sensors/add/ HTTP/1.1" 200 16170
"GET /technician/sensors/add/ HTTP/1.1" 200 15686
"GET /technician/sensors/add/ HTTP/1.1" 200 15628
"GET /technician/sensors/ HTTP/1.1" 200 10688
"GET /technician/sensors/add/ HTTP/1.1" 200 15628
"GET /technician/dashboard/ HTTP/1.1" 200 25674
- Broken pipe from ('127.0.0.1', 65141)
"GET /technician/soil-readings/ HTTP/1.1" 200 36250
C:\Users\user\Desktop\soils-sense\apps\technician\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\user\Desktop\soils-sense\apps\administrator\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\user\Desktop\soils-sense\apps\farmer\urls.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\user\Desktop\soils-sense\soilsense\settings.py changed, reloading.
Watching for file changes with StatReloader
"GET /technician/sensors/ HTTP/1.1" 302 0
"GET /authentication/login/?next=/technician/sensors/ HTTP/1.1" 200 3177
"POST /authentication/login/ HTTP/1.1" 302 0
"GET /administrator/ HTTP/1.1" 200 6752
"GET /administrator/users/ HTTP/1.1" 200 26416
Not Found: /media/users/profile_images/default.webp
"GET /media/users/profile_images/default.webp HTTP/1.1" 404 4235
"GET /static/images/default.webp HTTP/1.1" 304 0
"GET /administrator/server-logs/ HTTP/1.1" 200 20325
"GET /authentication/logout/ HTTP/1.1" 302 0
"GET /landing_page/ HTTP/1.1" 200 15668
"GET /static/images/dashboard.png HTTP/1.1" 404 1987
"GET /static/images/chart-preview.png HTTP/1.1" 404 1999
"GET /static/images/ladyone.jpg HTTP/1.1" 304 0
"GET /static/images/garden2.jpg HTTP/1.1" 304 0
"GET /static/images/lady.jpg HTTP/1.1" 304 0
"GET /static/images/lady3.jpg HTTP/1.1" 304 0
"GET /authentication/login/ HTTP/1.1" 200 2899
"POST /authentication/login/ HTTP/1.1" 302 0
"GET /farmer/dashboard/ HTTP/1.1" 200 25626
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/notifications/ HTTP/1.1" 200 14638
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/notifications/mark-read/2/ HTTP/1.1" 302 0
"GET /farmer/notifications/ HTTP/1.1" 200 14268
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/notifications/mark-read/1/ HTTP/1.1" 302 0
"GET /farmer/notifications/ HTTP/1.1" 200 13898
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/farm-management/ HTTP/1.1" 200 31686
- Broken pipe from ('127.0.0.1', 65482)
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/soil-data/ HTTP/1.1" 200 24310
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/api/sensor-data-json/6/?minutes=30 HTTP/1.1" 200 33
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/api/sensor-data-json/6/?minutes=30 HTTP/1.1" 200 33
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/api/sensor-data-json/6/?minutes=30 HTTP/1.1" 200 33
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/api/sensor-data-json/6/?minutes=30 HTTP/1.1" 200 33
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/api/sensor-data-json/6/?minutes=30 HTTP/1.1" 200 33
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/api/sensor-data-json/6/?minutes=30 HTTP/1.1" 200 33
"GET /authentication/logout/ HTTP/1.1" 302 0
"GET /landing_page/ HTTP/1.1" 200 15668
"GET /static/images/dashboard.png HTTP/1.1" 404 1987
"GET /static/images/chart-preview.png HTTP/1.1" 404 1999
"GET /authentication/login/ HTTP/1.1" 200 2899
"POST /authentication/login/ HTTP/1.1" 302 0
"GET /farmer/dashboard/ HTTP/1.1" 200 25632
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /authentication/logout/ HTTP/1.1" 302 0
"GET /landing_page/ HTTP/1.1" 200 15668
"GET /static/images/dashboard.png HTTP/1.1" 404 1987
"GET /static/images/chart-preview.png HTTP/1.1" 404 1999
"GET /authentication/login/ HTTP/1.1" 200 2899
"POST /authentication/login/ HTTP/1.1" 302 0
"GET /technician/dashboard/ HTTP/1.1" 200 25638
"GET /technician/sensors/ HTTP/1.1" 200 10374
"GET /technician/sensors/add/ HTTP/1.1" 200 15612
"POST /technician/sensors/add/ HTTP/1.1" 302 0
"GET /technician/sensors/ HTTP/1.1" 200 10721
Not Found: /farmer/get-soil-data/
"POST /farmer/get-soil-data/ HTTP/1.1" 404 45
"GET /authentication/logout/ HTTP/1.1" 302 0
"GET /landing_page/ HTTP/1.1" 200 15927
"GET /static/images/dashboard.png HTTP/1.1" 404 1987
"GET /static/images/chart-preview.png HTTP/1.1" 404 1999
"GET /authentication/login/ HTTP/1.1" 200 2899
"POST /authentication/login/ HTTP/1.1" 302 0
Not Found: /farmer/get-soil-data/
"POST /farmer/get-soil-data/ HTTP/1.1" 404 45
"GET /farmer/dashboard/ HTTP/1.1" 200 25628
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/soil-data/ HTTP/1.1" 200 24310
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/api/sensor-data-json/6/?minutes=30 HTTP/1.1" 200 33
Not Found: /farmer/get-soil-data/
"POST /farmer/get-soil-data/ HTTP/1.1" 404 45
"GET /farmer/farm-management/ HTTP/1.1" 200 31686
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/soil-data/ HTTP/1.1" 200 24310
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/api/sensor-data-json/6/?minutes=30 HTTP/1.1" 200 33
"GET /farmer/farm-management/ HTTP/1.1" 200 31686
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
Not Found: /farmer/get-soil-data/
"POST /farmer/get-soil-data/ HTTP/1.1" 404 45
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
C:\Users\user\Desktop\soils-sense\apps\administrator\urls.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\user\Desktop\soils-sense\apps\administrator\data_views.py changed, reloading.
Watching for file changes with StatReloader
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /authentication/logout/ HTTP/1.1" 302 0
"GET /landing_page/ HTTP/1.1" 200 16070
"GET /static/images/dashboard.png HTTP/1.1" 200 73583
"GET /static/images/chart.png HTTP/1.1" 200 69919
"GET /authentication/login/ HTTP/1.1" 200 2899
"POST /authentication/login/ HTTP/1.1" 302 0
"GET /technician/dashboard/ HTTP/1.1" 200 25656
"GET /technician/farm-locations/ HTTP/1.1" 200 16602
"GET /technician/dashboard/ HTTP/1.1" 200 25656
Not Found: /farmer/get-soil-data/
"POST /farmer/get-soil-data/ HTTP/1.1" 404 45
"GET /authentication/logout/ HTTP/1.1" 302 0
"GET /landing_page/ HTTP/1.1" 200 16070
"GET /static/images/chart.png HTTP/1.1" 304 0
"GET /static/images/dashboard.png HTTP/1.1" 304 0
"GET /authentication/login/ HTTP/1.1" 200 2899
"POST /authentication/login/ HTTP/1.1" 302 0
"GET /administrator/ HTTP/1.1" 200 6752
"GET /administrator/data/ HTTP/1.1" 200 30972
Not Found: /media/users/profile_images/default.webp
"GET /media/users/profile_images/default.webp HTTP/1.1" 404 4235
Not Found: /farmer/get-soil-data/
"POST /farmer/get-soil-data/ HTTP/1.1" 404 45
Not Found: /farmer/get-soil-data/
"POST /farmer/get-soil-data/ HTTP/1.1" 404 45
Not Found: /farmer/get-soil-data/
"POST /farmer/get-soil-data/ HTTP/1.1" 404 45
Not Found: /farmer/get-soil-data/
"POST /farmer/get-soil-data/ HTTP/1.1" 404 45
Not Found: /farmer/get-soil-data/
"POST /farmer/get-soil-data/ HTTP/1.1" 404 45
"GET /administrator/data/soil-moisture/73/ HTTP/1.1" 200 18731
Not Found: /farmer/get-soil-data/
"POST /farmer/get-soil-data/ HTTP/1.1" 404 45
"GET /administrator/data/ HTTP/1.1" 200 30972
Not Found: /media/users/profile_images/default.webp
"GET /media/users/profile_images/default.webp HTTP/1.1" 404 4235
Watching for file changes with StatReloader
"GET / HTTP/1.1" 200 16070
"GET /static/images/dashboard.png HTTP/1.1" 304 0
"GET /static/images/chart.png HTTP/1.1" 304 0
"GET /authentication/signup/ HTTP/1.1" 200 4168
"GET /authentication/login/ HTTP/1.1" 200 2899
"POST /authentication/login/ HTTP/1.1" 302 0
"GET /administrator/ HTTP/1.1" 200 6752
"GET /administrator/ml-models/ HTTP/1.1" 200 27036
"GET /administrator/ml-models/irrigation_recommendation/?algorithm=gradient_boosting HTTP/1.1" 200 27644
"GET /administrator/users/ HTTP/1.1" 200 33491
"GET /static/images/default.webp HTTP/1.1" 304 0
Not Found: /media/users/profiles/default.webp
"GET /media/users/profiles/default.webp HTTP/1.1" 404 4211
Not Found: /media/users/profiles/default.webp
"GET /media/users/profiles/default.webp HTTP/1.1" 404 4211
"GET /administrator/ HTTP/1.1" 200 6752
"GET /administrator/users/ HTTP/1.1" 200 33491
Not Found: /media/users/profiles/default.webp
"GET /media/users/profiles/default.webp HTTP/1.1" 404 4211
Not Found: /media/users/profiles/default.webp
"GET /media/users/profiles/default.webp HTTP/1.1" 404 4211
"GET /administrator/data/ HTTP/1.1" 200 43704
Not Found: /media/users/profiles/default.webp
"GET /media/users/profiles/default.webp HTTP/1.1" 404 4211
Not Found: /media/users/profiles/default.webp
"GET /media/users/profiles/default.webp HTTP/1.1" 404 4211
"GET /administrator/data/farms/3/ HTTP/1.1" 200 15322
"GET /authentication/logout/ HTTP/1.1" 302 0
"GET /landing_page/ HTTP/1.1" 200 16070
"GET /static/images/chart.png HTTP/1.1" 304 0
"GET /static/images/dashboard.png HTTP/1.1" 304 0
"GET /authentication/login/ HTTP/1.1" 200 2899
"POST /authentication/login/ HTTP/1.1" 302 0
"GET /farmer/dashboard/ HTTP/1.1" 200 24774
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/soil-data/ HTTP/1.1" 200 24318
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/api/sensor-data-json/3/?minutes=30 HTTP/1.1" 200 33
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/api/sensor-data-json/3/?minutes=30 HTTP/1.1" 200 33
"POST /farmer/get-soil-data/ HTTP/1.1" 200 175
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/api/sensor-data-json/3/?minutes=30 HTTP/1.1" 200 226
"POST /farmer/get-soil-data/ HTTP/1.1" 200 169
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/api/sensor-data-json/3/?minutes=30 HTTP/1.1" 200 415
"POST /farmer/get-soil-data/ HTTP/1.1" 200 166
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/api/sensor-data-json/3/?minutes=30 HTTP/1.1" 200 601
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"POST /farmer/get-soil-data/ HTTP/1.1" 200 176
"GET /farmer/api/sensor-data-json/3/?minutes=30 HTTP/1.1" 200 797
"GET /farmer/download-sensor-data/?farm_id=3 HTTP/1.1" 200 369
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"POST /farmer/get-soil-data/ HTTP/1.1" 200 175
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"POST /farmer/get-soil-data/ HTTP/1.1" 200 166
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"POST /farmer/get-soil-data/ HTTP/1.1" 200 169
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"POST /farmer/get-soil-data/ HTTP/1.1" 200 169
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"POST /farmer/get-soil-data/ HTTP/1.1" 200 166
"GET /farmer/predictions/ HTTP/1.1" 200 27266
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/analytics/ HTTP/1.1" 200 21781
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"POST /farmer/get-soil-data/ HTTP/1.1" 200 176
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/recommendations/ HTTP/1.1" 200 17749
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"GET /farmer/profile/ HTTP/1.1" 200 15285
"GET /farmer/notifications/unread-count/ HTTP/1.1" 200 12
"POST /farmer/get-soil-data/ HTTP/1.1" 200 175
"GET /authentication/logout/ HTTP/1.1" 302 0
"GET /landing_page/ HTTP/1.1" 200 16070
"GET /static/images/dashboard.png HTTP/1.1" 304 0
"GET /static/images/chart.png HTTP/1.1" 304 0
- Broken pipe from ('127.0.0.1', 49693)
"GET /static/images/ladyone.jpg HTTP/1.1" 304 0
"GET /authentication/login/ HTTP/1.1" 200 2899
"POST /authentication/login/ HTTP/1.1" 302 0
"GET /technician/dashboard/ HTTP/1.1" 200 25646
"POST /farmer/get-soil-data/ HTTP/1.1" 200 165
"POST /farmer/get-soil-data/ HTTP/1.1" 200 174
"POST /farmer/get-soil-data/ HTTP/1.1" 200 176
"GET /technician/reports/ HTTP/1.1" 200 41893
"POST /farmer/get-soil-data/ HTTP/1.1" 200 176
"GET /technician/profile/ HTTP/1.1" 200 19832
"GET /technician/settings/ HTTP/1.1" 200 17590
"GET /technician/dashboard/ HTTP/1.1" 200 25648
"POST /farmer/get-soil-data/ HTTP/1.1" 200 166
"GET /technician/models/ HTTP/1.1" 200 37686
"POST /farmer/get-soil-data/ HTTP/1.1" 200 169
"POST /farmer/get-soil-data/ HTTP/1.1" 200 166
"POST /farmer/get-soil-data/ HTTP/1.1" 200 176
"POST /farmer/get-soil-data/ HTTP/1.1" 200 176
"POST /farmer/get-soil-data/ HTTP/1.1" 200 175
"POST /farmer/get-soil-data/ HTTP/1.1" 200 165
"POST /farmer/get-soil-data/ HTTP/1.1" 200 166
"POST /farmer/get-soil-data/ HTTP/1.1" 200 175
"POST /farmer/get-soil-data/ HTTP/1.1" 200 175
"GET /authentication/logout/ HTTP/1.1" 302 0
"GET /landing_page/ HTTP/1.1" 200 16070
"GET /static/images/chart.png HTTP/1.1" 304 0
"GET /static/images/dashboard.png HTTP/1.1" 304 0
"GET /authentication/login/ HTTP/1.1" 200 2899
"POST /farmer/get-soil-data/ HTTP/1.1" 200 176
"POST /authentication/login/ HTTP/1.1" 302 0
"GET /administrator/ HTTP/1.1" 200 6752
- Broken pipe from ('127.0.0.1', 49821)
C:\Users\user\Desktop\soils-sense\apps\administrator\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\user\Desktop\soils-sense\apps\administrator\templatetags\administrator.py changed, reloading.
Watching for file changes with StatReloader
Watching for file changes with StatReloader
"GET / HTTP/1.1" 200 16070
"GET /authentication/login/ HTTP/1.1" 200 2899
"POST /authentication/login/ HTTP/1.1" 302 0
"GET /administrator/ HTTP/1.1" 200 16703
"GET /administrator/sensors/ HTTP/1.1" 200 14004
"GET /administrator/notifications/ HTTP/1.1" 200 16240
"GET /administrator/sensors/ HTTP/1.1" 200 14004
"GET /administrator/sensors/3/ HTTP/1.1" 200 13488
"GET /administrator/data/ HTTP/1.1" 200 43679
Not Found: /media/users/profiles/default.webp
"GET /media/users/profiles/default.webp HTTP/1.1" 404 4211
Not Found: /media/users/profiles/default.webp
"GET /media/users/profiles/default.webp HTTP/1.1" 404 4211
Watching for file changes with StatReloader
"GET / HTTP/1.1" 200 16070
"GET /static/images/dashboard.png HTTP/1.1" 304 0
"GET /static/images/chart.png HTTP/1.1" 304 0
"GET /static/images/ladyone.jpg HTTP/1.1" 304 0
Not Found: /favicon.ico
"GET /favicon.ico HTTP/1.1" 404 4081
"GET /static/images/garden2.jpg HTTP/1.1" 304 0
Not Found: /privacy
"GET /privacy HTTP/1.1" 404 4069
"GET /static/images/lady.jpg HTTP/1.1" 304 0
"GET /static/images/lady3.jpg HTTP/1.1" 304 0
"GET /authentication/login/ HTTP/1.1" 200 2899
"POST /authentication/login/ HTTP/1.1" 302 0
"GET /technician/dashboard/ HTTP/1.1" 200 26149
"GET /technician/models/ HTTP/1.1" 200 37686
C:\Users\user\Desktop\soils-sense\apps\administrator\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\user\Desktop\soils-sense\ml\__init__.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\user\Desktop\soils-sense\soilsense\settings.py changed, reloading.
Watching for file changes with StatReloader
"GET / HTTP/1.1" 200 19375
"GET /authentication/signup/ HTTP/1.1" 200 4168
Not Found: /privacy
"GET /privacy HTTP/1.1" 404 4069
"GET /authentication/login/ HTTP/1.1" 200 2899
"GET / HTTP/1.1" 200 19185
"GET / HTTP/1.1" 200 19246
"GET / HTTP/1.1" 200 19246
"GET / HTTP/1.1" 200 19246
"GET / HTTP/1.1" 200 19246
"GET / HTTP/1.1" 200 19246
C:\Users\user\Desktop\soils-sense\apps\landing_page\views.py changed, reloading.
Watching for file changes with StatReloader
"GET / HTTP/1.1" 200 19375
"GET /privacy HTTP/1.1" 301 0
"GET /privacy/ HTTP/1.1" 200 11745
"GET /terms HTTP/1.1" 301 0
"GET /terms/ HTTP/1.1" 200 11944
"GET /privacy/ HTTP/1.1" 200 11745
"GET /terms/ HTTP/1.1" 200 11944
"GET / HTTP/1.1" 200 19375
"GET /privacy/ HTTP/1.1" 200 11745
"GET /terms/ HTTP/1.1" 200 11944
"GET /authentication/login/ HTTP/1.1" 200 2899
"POST /authentication/login/ HTTP/1.1" 302 0
"GET /technician/dashboard/ HTTP/1.1" 200 26147
"GET /technician/sensors/ HTTP/1.1" 200 11379
"GET /technician/sensors/add/ HTTP/1.1" 200 15680
"GET /technician/profile/ HTTP/1.1" 200 19832
"GET /technician/settings/ HTTP/1.1" 200 17590
"GET /technician/models/ HTTP/1.1" 200 54632
C:\Users\user\Desktop\soils-sense\apps\technician\views.py changed, reloading.
Watching for file changes with StatReloader
"GET /technician/models/ HTTP/1.1" 200 54632
"GET /technician/models/ HTTP/1.1" 200 54632
"GET /technician/dashboard/ HTTP/1.1" 200 26149
"GET /technician/models/ HTTP/1.1" 200 54632
C:\Users\user\Desktop\soils-sense\apps\technician\views.py changed, reloading.
Watching for file changes with StatReloader
"GET /technician/models/ HTTP/1.1" 200 54632
"GET /technician/sensors/ HTTP/1.1" 200 11379
"GET /technician/sensors/add/ HTTP/1.1" 200 15680
"GET /technician/sensors/ HTTP/1.1" 200 11379
"GET /technician/soil-readings/ HTTP/1.1" 200 36257
"GET /technician/soil-readings/ HTTP/1.1" 200 36257
"GET /technician/analytics/ HTTP/1.1" 200 17635
"GET /technician/reports/ HTTP/1.1" 200 58091
"GET /technician/prediction/14/download-pdf/ HTTP/1.1" 200 2748
"GET /technician/analytics/ HTTP/1.1" 200 17635
"GET /technician/dashboard/ HTTP/1.1" 200 26149
C:\Users\user\Desktop\soils-sense\apps\landing_page\views.py changed, reloading.
Watching for file changes with StatReloader
